---
title: "Efficient tools for analyzing genome-wide data of clonal organisms"
runtitle: "Efficient clones"
author:
  - name: Zhian N. Kamvar
    affiliation: '1'
    etal: Kamvar
  - name: Jonah C. Brooks
    affiliation: '2'
  - name: Niklaus J. Grunwald
    affiliation: '1,3'
    email: grunwaln@science.oregonstate.edu
    institution: Horticultural Crops Research Laboratory USDA ARS
    street: 3420 NW Orchard Ave.
    city: Corvallis
    state: OR
    zip: 97330
affiliation:
  - id: '1'
    department: Botany and Plant Pathology
    institution: Oregon State University
    city: Corvallis
    state: OR
    country: USA
  - id: '2' 
    department: College of Electrical Engineering and Computer Science
    institution: Oregon State University
    city: Corvallis
    state: OR
    country: USA
  - id: '3' 
    department: Horticultural Crops Research Laboratory
    institution: USDA-Agricultural Research Service
    city: Corvallis
    state: OR
    country: USA
output: FrontiersTemplate::frontiers_article
csl: frontiers.csl
bibliography: citations.bib
---


\begin{abstract}

Understanding the dynamics of plant pathogen populations is imperative to
understanding plant microbe interactions, adaptation of pathogens to hosts,
reactive breeding and deployment of R genes and design of informed management
strategies. With the advent of high throughput sequencing technologies,
obtaining genomic sequences for representative populations has been easier than
ever before. This in combination with a move towards more reproducible research
has led researchers to move away from traditional stand alone programs for
population genetic analysis to modular programs such as R, which has a vibrant
community of contributers submitting packages tailored for several analyses,
including population genetics. The R package poppr was released to specifically
address issues surrounding analysis of clonal populations and has recently
implemented several tools that better facilitate the analysis of clonal
populations across different levels of hierarchies, implementing multilocus
genotype definitions and the index of association for reduced representation
genomic data, and modular bootstrapping of any genetic distance.

\tiny
 \keyFont{ \section{Keywords:} clonality, population genetics, bootstrap, open source}
\end{abstract}


# Introduction {-}

- Rise in popularity of reproducible methods | Should this go in discussion? It is one great advantage of using R!
    - shitty standalone programs 

Most computational tools for population genetics are based on concepts developed
for sexual model organisms. Populations that reproduce clonally or are polyploid
are thus difficult to characterize using classical population genetic tools.

The research community using the R statistical and computing language [@R] has
developed a plethora of new resources for population genetic analysis [@pegas;
@Jombart_2008]. Recently, we introduced the R package *poppr* specifically
developed for analysis of clonal populations [@kamvar2014poppr]. *Poppr*
previously introduced several novel features including the ability to conduct a
hierarchical analysis across unlimited hierarchies, test for linkage
association, graph minimum spanning networks or provide bootstrap support for
Bruvo's distance in resulting trees. In its first iteration, *poppr* was
appropriate for traditional markers systems, but not ideally suited to
population genomic data resulting from high throughput sequencing methods as
increases in the number of markers decreased the ability to distinguish clonal
genotypes due to higher chance to detect somatic mutations or genotyping error.

- Packages available in R
    - Previous lack of tools for clonal populations, still lack of tools for genomic data sets
        - How do you deal with clones?
        - How do you detect linkage? 
        - How do you know if you've sampled enough?
        
Multiple references get a semi-colon [@wilson2014best; @wilson2014best]. References that are simply mentioned, such as @wilson2014best, are cited without square braces.


# Materials and Methods {-}

In *poppr* version 1.1, a new S4 object was defined to expand the genind object
of *adegenet*. The genclone object formalized the definitions of multilocus
genotypes and population hierarchies by adding two slots called mlg and
hierarchy that carried a numeric vector and a data frame, respectively. These
new slots allow for increased efficiency and ease of use by allowing these
metadata to travel with the genetic data.

## Clone correction {-}

As highlighted in previous work, clone correction is an important part of
population genetic analysis of organisms that have cryptic growth or are known
to produce asexual propagules [@kamvar2014poppr; @milgroom1996recombination;
@grunwald2003analysis]. This method removes bias that would otherwise affect
metrics that rely on allele frequencies. It was initially designed for data with
only a handful of markers. With the advent of large-scale sequencing and
reduced- representation libraries, it has become easier to sequence tens of
thousands of markers from hundreds of individuals [@elshire2011robust;
@davey2011genome; @davey2010rad]. With this number of markers, the genetic
resolution is much greater, but the chance of genotyping error is also greatly
increased [@mastretta2014gene]. Taking this and somatic mutation into account,
it would be impossible to separate true clones from independent individuals by
just comparing what multilocus genotypes are different. We introduce a a new
transparent method for collapsing unique multilocus genotypes determined by
naive string comparison into multilocus lineages utilizing any genetic distance
given three different clustering algorithms: farthest neighbor, nearest
neighbor, and UPGMA (average neighbor) [citation needed].

The clustering algorithms act on a distance matrix that is either provided by
the user or generated via a function that will calculate a distance from
genclone objects such as `bruvo.dist`, which will calculate Bruvo's distance at
any level of ploidy. All algorithms have been implemented in C and utilize the
open MP framework for optional parallel processing. Default is the conservative
farthest neighbor algorithm, which will only cluster samples together only if
all samples in the cluster are at a distance less than the given threshold. By
contrast, the nearest neighbor algorithm will have a chaining effect that will
cluster samples akin to adding links on a chain where a sample can be included
in a cluster if all of the samples have at least one connection below a given
threshold. The UPGMA, or average neighbor clustering algorithm is the one most
familiar to biologists as it is often used to generate preliminary ultra-metric
trees based off of genetic distance. This algorithm will cluster by creating a
representative sample per cluster and joining clusters if these representative
samples are closer than the given threshold.

### *P. infestans* {-}

*Phytophthora infestans* is the causal agent of potato late blight originating
from Mexico and spread to Europe in the mid 19th century [@goss2014irish; cooke].
*P. infestans* reproduces both clonally and sexually. The clonal lineages of *P.
infestans* have been formally defined into 18 separate clonal lineages using
various molecular methods. We utilize these data to show how the `mlg.filter`
function collapses multilocus genotypes with Bruvo's distance assuming a genome
addition model [@bruvo2004simple]. 


## Index of association {-}

The index of association ($I_A$) is a measure of multilocus linkage
disequilibrium that is most often used to detect clonal reproduction within
organisms that have the ability to reproduce via sexual or asexual processes
[brown1980; maynard1993; @milgroom1996recombination]. It was standardized in
2001 as $\bar{r}_d$ by [@Agapow_2001] to address the issue of scaling with
increasing number of loci. This metric is typically applied to traditional
dominant and co-dominant markers such as AFLPs, SNPs, or microsatellite markers.
With the advent of high throughput sequencing, SNP data is now available in in a
genome-wide context and in very large matrices including thousands of SNPs.
Thus, the likelihood of finding mutations within two individuals of a given
clone increases and tools are needed for defining clone boundaries. For this
reason, we  devised two approaches using the index of association for large
numbers of markers typical for population genomic studies. 

The first approach is a sliding window approach that would utilize the position
of markers in the genome to calculate $\bar{r}_d$ among any number of SNPs found
within the windowed region. It is important that this calculation utilize
$\bar{r}_d$ as the number of loci will be different within each window
[@Agapow_2001]. This approach would be suited for a quick calculation of linkage
disequilibrium across the genome that can detect potential hotspots of LD that
could be investigated further with more computationally intensive methods
assuming that the number of samples << the number of loci.

A sliding window approach would not be good for utilizing $\bar{r}_d$ as a test
for clonal reproduction as it would necessarily focus on loci within a short
section of the genome that may or may not be recombining. A remedy for this is
to randomly sample $m$ loci, calculate $\bar{r}_d$ and repeat $r$ times,
creating a distribution of expected values of $\bar{r}_d$. 

As the data for this would be stored in a genlight object, which stores genetic
information in bits, the calculations of $\bar{r}_d$ are performed in
parallelized bit level C code for efficiency [@Jombart_2008].

 - Clone correction based on genetic distance
    - reduced representation data (GBS)
    - consistent clones with one or two highly variable loci (Cooke example)
    - nearest neighbor; farthest neighbor; average neighbor
 - Index of Association
    - Invented for small number of markers
    - Methods of $\bar{r}_d$ for large number of markers:
      - sliding window (could be a fast way to detect linkage)
      - random sampling of x loci (creates a distribution of $\bar{r}_d$)
 - Bootstrapping genetic distance
    - Matrix algebra makes it easy to calculate genetic distance in R (with alleles in columns).
    - Alleles in columns makes it hard to bootstrap (alleles are not independent units).
 - Population Stratification
    - Necessary for analyses of factorial sampling [@everhart2014fine] or hierarchical sampling [@linde2002population]
    - Implementation in StrataG and Hierfstat loosely defined.
    - Solution in S4 classes: strict definition.
    - adopted by larger community
    

## Genotype Accumulation Curve {-}

Analysis of population genetics of clonal organisms often borrows from
ecological methods such as analysis of diversity within populations
[@milgroom1996recombination; @arnaud2007standardizing]. When choosing markers
for analysis, it is important to make sure that the observed diversity in your
sample will not appreciably increase if an additional marker is added
[@arnaud2007standardizing]. This concept is analogous to a species accumulation
curve, obtained by rarefaction. The genotype accumulation curve in *poppr* is
implemented by randomly sampling $x$ loci and counting the number of observed
MLGs. This repeated $r$ times for 1 locus up to $n-1$ loci, creating $n-1$
distributions of observed MLGs.

## Minimum Spanning Networks {-}

Poppr introduced minimum spanning networks in it's original iteration that were
based off of igraph's function minimum.spanning.tree. This algorithm would
produce a minimum spanning tree, but with no reticulations, which is common and
expected of clonal organisms. In other minimum spanning network programs,
reticulation is obtained by caluclating the minimum spanning tree several times
and returning the set of all edges included in the trees. Due to the way igraph
has implemented Prim's algorithm, it is not possible to utilize this stategy,
thus we have implemented a C function to walk through a distance matrix and
minimum spanning tree to connect groups of nodes with edges of equal weight.
These reticulate minimum spanning networks are very important for extremely
clonal organisms where a minimum spanning tree would become a chain. The current
implementation been successfuly used in analyses such as reconstruction of the
*P. ramorum* outbreak in Curry County, OR [kamvar2015spatial]. Reticulated
networks also allow for the addition of graph community detection algorithms
such as the infomap algorithm, giving the user another tool to detect clonal or
population structure [@rosvall2008maps].

## Additions {-}

- Psex
- Bootstrapping for individuals or populations for any genetic distance
- MLG clustering
- Index of Association for genomic sequences
- Parallelization
- Genotype Accumulation Curve
- Matching Ties in Minimum Spanning networks (Cite ramorum paper)

# Results {-}

## Clonal identification {-}

### *P. infestans* {-}

The three algorithms were able to detect 18 multilocus lineages at different
distances (Fig. 1). Contingency tables between the described multilocus
genotypes and the genotypes defined by distance show that most of the 18
lineages were resolved, except for US-8, which is polytomic (Table 1).

```{r, pinf_data, echo = FALSE, message = FALSE}
source("../Rscripts/my_functions.R")
library('poppr')
library('ape')
library('phangorn')
library('animation')
infdat <- RCurl::getURL("https://raw.githubusercontent.com/grunwaldlab/phytophthora_id/master/shiny-server/www/genoid_infestans/reduced_database.txt.csv")
infdat <- read.table(text = infdat, head = TRUE)
pinf   <- df2genind(infdat[-c(1,2)], sep = "/", ploidy = 3, ind.names = infdat[[1]], pop = infdat[[2]])
ssr    <- c(3, 3, 2, 3, 3, 2, 2, 3, 3, 3, 3, 3)
x      <- as.genclone(pinf)
fstats <- filter_stats(x, bruvo.dist, plot = TRUE, replen = ssr, loss = FALSE, nclone = 18)
title(main = expression(paste(italic("P. infestans"), " reference isolates (12 SSR loci)")))
thresh18 <- fstats$average$thresholds[nInd(x) - 18] + .Machine$double.eps^0.5
z <- mlg.filter(x, threshold = thresh18, distance = bruvo.dist, replen = ssr, loss = FALSE, stats = "MLGS")
```
```{r, pinf_tree, echo = FALSE, message = FALSE, fig.height = 10, fig.width = 5}
pinfupgma <- upgma(bruvo.dist(x, replen = ssr, loss = FALSE))
color_mlg_tree(x, pinfupgma, z)
axisPhylo(3)
```
```{r pinf_table, echo = FALSE, message = FALSE, results='asis'}
library("xtable")

res <- table(pop(x), z, dnn = NULL)
res[res == 0] <- "."
print.xtable(xtable(res, align = rep("c", ncol(res) + 1)), comment = FALSE)
```

## Speed {-}

## Community Acceptance {-}

- articles that have cited poppr
- hierarchical methods have been ported to adegenet

# Discussion {-}

- loops in graphs allow for analysis of population structure via graph walking algorithms in igraph
- bootstrapping methods encourage future developers to write distance implementations in common format
- moving towards open source, modular tools is the direction that population genetics and plant pathology needs to go. 




# References {-}