%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% This is just an example/guide for you to refer to when submitting manuscripts to Frontiers, it is not mandatory to use frontiers.cls nor frontiers.tex  %
% This will only generate the Manuscript, the final article will be typeset by Frontier after acceptance.                                                 %
%                                                                                                                                                         %
% When submitting your files, remember to upload this *tex file, the pdf generated with it, the *bib file (if bibliography is not within the *tex) and all the figures.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% Version 3.0 Generated 2014/12/19 %%%
%%% You will need to have the following packages installed: datetime, fmtcount, etoolbox, fcprefix, which are normally inlcuded in WinEdt. %%%
%%% In http://www.ctan.org/ you can find the packages and how to install them, if necessary. %%%

\documentclass{frontiersSCNS} % for Science, Engineering and Humanities and Social Sciences articles
%\documentclass{frontiersHLTH} % for Health articles
%\documentclass{frontiersFPHY} % for Physics articles

%\setcitestyle{square}
\usepackage{url,lineno}
\linenumbers


% BELOW TAKEN FROM rticles plos template
%
% amsmath package, useful for mathematical formulas
\usepackage{amsmath}
% amssymb package, useful for mathematical symbols
\usepackage{amssymb}

% hyperref package, useful for hyperlinks
\usepackage{hyperref}

% graphicx package, useful for including eps and pdf graphics
% include graphics with the command \includegraphics
\usepackage{graphicx}

% Sweave(-like)
\usepackage{fancyvrb}
\DefineVerbatimEnvironment{Sinput}{Verbatim}{fontshape=sl}
\DefineVerbatimEnvironment{Soutput}{Verbatim}{}
\DefineVerbatimEnvironment{Scode}{Verbatim}{fontshape=sl}
\newenvironment{Schunk}{}{}
\DefineVerbatimEnvironment{Code}{Verbatim}{}
\DefineVerbatimEnvironment{CodeInput}{Verbatim}{fontshape=sl}
\DefineVerbatimEnvironment{CodeOutput}{Verbatim}{}
\newenvironment{CodeChunk}{}{}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

\usepackage{color}

% Below is from frontiers
%
\bibliographystyle{frontiersinSCNS}
% Use doublespacing - comment out for single spacing
%\usepackage{setspace}
%\doublespacing


% Leave a blank line between paragraphs instead of using \\



\def\keyFont{\fontsize{8}{11}\helveticabold }

%% ** EDIT HERE **
%% PLEASE INCLUDE ALL MACROS BELOW

%% END MACROS SECTION



  \def\Authors{
  Zhian N. Kamvar\,\textsuperscript{1},
  Jonah C. Brooks\,\textsuperscript{2},
  Niklaus J. Grunwald\,\textsuperscript{1,3*}}
% \\
\def\Address{
  \textsuperscript{1} Botany and Plant Pathology, Oregon State University,  Corvallis,  OR,  USA\\
  \textsuperscript{2} College of Electrical Engineering and Computer Science, Oregon State University,  Corvallis,  OR,  USA\\
  \textsuperscript{3} Horticultural Crops Research Laboratory, USDA-Agricultural Research Service,  Corvallis,  OR,  USA\\
}

  
  \def\firstAuthorLast{Kamvar {et~al.}}
  
  
  \def\corrAuthor{Niklaus J. Grunwald}\def\corrAddress{Horticultural Crops Research Laboratory USDA ARS\\3420 NW Orchard Ave.\\Corvallis, OR, 97330}\def\corrEmail{\href{mailto:grunwaln@science.oregonstate.edu}{\nolinkurl{grunwaln@science.oregonstate.edu}}}
  


\begin{document}
\onecolumn
\firstpage{1}

\title[Reproducible clones]{Reproducible research for clonal organisms}
\author[\firstAuthorLast]{\Authors}
\address{}
\correspondance{}
\extraAuth{}% If there are more than 1 corresponding author, comment this line and uncomment the next one.
%\extraAuth{corresponding Author2 \\ Laboratory X2, Institute X2, Department X2, Organization X2, Street X2, City X2 , State XX2 (only USA, Canada and Australia), Zip Code2, X2 Country X2, email2@uni2.edu}
\topic{}% If your article is part of a Research Topic, please indicate here which.
\maketitle

\begin{abstract}

Understanding the dynamics of plant pathogen populations is imperative to
understanding what management strategies are important. With the advent of more
advanced and cheaper sequencing technologies, obtaining genomic sequences for
representative populations has been easier than ever before. This in combination
with a move towards more reproducible research has lead researchers to move away
from traditional stand alone programs for population genetic analysis to modular
programs such as R, which has a vibrant community of contributers submitting
packages tailored for several analyses, including population genetics. The R
package poppr was released to specifically address issues surrounding analysis
of clonal populations and has recently implemented several tools that better
facilitate the analysis of clonal populations across different levels of
hierarchies, implementing multilocus genotype definitions and the index of
association for reduced representation genomic data, and modular bootstrapping
of any genetic distance.

\tiny
 \keyFont{ \section{Keywords:} clonality population genetics bootstrap open source}
\end{abstract}

\section*{Introduction}\label{introduction}
\addcontentsline{toc}{section}{Introduction}

Multiple references get a semi-colon (Wilson et al., 2014, 2014).
References that are simply mentioned, such as Wilson et al. (2014), are
cited without square braces.

\begin{itemize}
\itemsep1pt\parskip0pt\parsep0pt
\item
  Rise in popularity of reproducible methods

  \begin{itemize}
  \itemsep1pt\parskip0pt\parsep0pt
  \item
    shitty standalone programs
  \end{itemize}
\item
  Packages available in R

  \begin{itemize}
  \itemsep1pt\parskip0pt\parsep0pt
  \item
    Previous lack of tools for clonal populations, still lack of tools
    for genomic data sets

    \begin{itemize}
    \itemsep1pt\parskip0pt\parsep0pt
    \item
      How do you deal with clones?
    \item
      How do you detect linkage?
    \item
      How do you know if you've sampled enough?
    \end{itemize}
  \end{itemize}
\end{itemize}

\section*{Materials and Methods}\label{materials-and-methods}
\addcontentsline{toc}{section}{Materials and Methods}

\subsection*{Clone correction}\label{clone-correction}
\addcontentsline{toc}{subsection}{Clone correction}

As highlighted in previous work, clone correction is an important part
of population genetic analysis of organisms that have cryptic growth or
are known to produce asexual propagules (Kamvar et al., 2014; Milgroom,
1996; Grünwald et al., 2003).

\begin{itemize}
\itemsep1pt\parskip0pt\parsep0pt
\item
  Clone correction based on genetic distance

  \begin{itemize}
  \itemsep1pt\parskip0pt\parsep0pt
  \item
    reduced representation data (GBS)
  \item
    consistent clones with one or two highly variable loci (Cooke
    example)
  \item
    nearest neighbor; farthest neighbor; average neighbor
  \end{itemize}
\item
  Index of Association

  \begin{itemize}
  \itemsep1pt\parskip0pt\parsep0pt
  \item
    Invented for small number of markers
  \item
    Methods of \(\bar{r}_d\) for large number of markers:
  \item
    sliding window (could be a fast way to detect linkage)
  \item
    random sampling of x loci (creates a distribution of \(\bar{r}_d\))
  \end{itemize}
\item
  Bootstrapping genetic distance

  \begin{itemize}
  \itemsep1pt\parskip0pt\parsep0pt
  \item
    Matrix algebra makes it easy to calculate genetic distance in R
    (with alleles in columns).
  \item
    Alleles in columns makes it hard to bootstrap (alleles are not
    independent units).
  \end{itemize}
\item
  Population Stratification

  \begin{itemize}
  \itemsep1pt\parskip0pt\parsep0pt
  \item
    Necessary for analyses of factorial sampling (Everhart and Scherm,
    2014) or hierarchical sampling (Linde et al., 2002)
  \item
    Implementation in StrataG and Hierfstat loosely defined.
  \item
    Solution in S4 classes: strict definition.
  \item
    adopted by larger community
  \end{itemize}
\end{itemize}

\subsection*{Additions}\label{additions}
\addcontentsline{toc}{subsection}{Additions}

\begin{itemize}
\itemsep1pt\parskip0pt\parsep0pt
\item
  Psex
\item
  Bootstrapping for individuals or populations for any genetic distance
\item
  MLG clustering
\item
  Index of Association for genomic sequences
\item
  Parallelization
\item
  Genotype Accumulation Curve
\item
  Matching Ties in Minimum Spanning networks (Cite ramorum paper)
\end{itemize}

\section*{Results}\label{results}
\addcontentsline{toc}{section}{Results}

\subsection*{Speed}\label{speed}
\addcontentsline{toc}{subsection}{Speed}

\subsection*{Community Acceptance}\label{community-acceptance}
\addcontentsline{toc}{subsection}{Community Acceptance}

\begin{itemize}
\itemsep1pt\parskip0pt\parsep0pt
\item
  articles that have cited poppr
\item
  hierarchical methods have been ported to adegenet
\end{itemize}

\section*{Discussion}\label{discussion}
\addcontentsline{toc}{section}{Discussion}

\begin{itemize}
\itemsep1pt\parskip0pt\parsep0pt
\item
  loops in graphs allow for analysis of population structure via graph
  walking algorithms in igraph
\item
  bootstrapping methods encourage future developers to write distance
  implementations in common format
\item
  moving towards open source, modular tools is the direction that
  population genetics and plant pathology needs to go.
\end{itemize}

\section*{References}\label{references}
\addcontentsline{toc}{section}{References}

Everhart, S., and Scherm, H. (2014). Fine-scale genetic structure of
monilinia fructicola during brown rot epidemics within individual peach
tree canopies. \emph{Phytopathology}.

Grünwald, N. J., Goodwin, S. B., Milgroom, M. G., and Fry, W. E. (2003).
Analysis of genotypic diversity data for populations of microorganisms.
\emph{Phytopathology} 93, 738--46. Available at:
\url{http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO.2003.93.6.738}.

Kamvar, Z. N., Tabima, J. F., and Gr{ü}nwald, N. J. (2014). Poppr: An r
package for genetic analysis of populations with clonal, partially
clonal, and/or sexual reproduction. \emph{PeerJ} 2, e281.

Linde, C., Zhan, J., and McDonald, B. (2002). Population structure of
mycosphaerella graminicola: From lesions to continents.
\emph{Phytopathology} 92, 946--955.

Milgroom, M. G. (1996). Recombination and the multilocus structure of
fungal populations. \emph{Annual Review of Phytopathology} 34, 457--477.

Wilson, G., Aruliah, D., Brown, C. T., Hong, N. P. C., Davis, M., Guy,
R. T., Haddock, S. H., Huff, K. D., Mitchell, I. M., Plumbley, M. D., et
al. (2014). Best practices for scientific computing. \emph{PLoS biology}
12, e1001745.


\end{document}

