%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% This is just an example/guide for you to refer to when submitting manuscripts to Frontiers, it is not mandatory to use frontiers.cls nor frontiers.tex  %
% This will only generate the Manuscript, the final article will be typeset by Frontier after acceptance.                                                 %
%                                                                                                                                                         %
% When submitting your files, remember to upload this *tex file, the pdf generated with it, the *bib file (if bibliography is not within the *tex) and all the figures.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% Version 3.0 Generated 2014/12/19 %%%
%%% You will need to have the following packages installed: datetime, fmtcount, etoolbox, fcprefix, which are normally inlcuded in WinEdt. %%%
%%% In http://www.ctan.org/ you can find the packages and how to install them, if necessary. %%%

\documentclass{frontiersSCNS} % for Science, Engineering and Humanities and Social Sciences articles
%\documentclass{frontiersHLTH} % for Health articles
%\documentclass{frontiersFPHY} % for Physics articles

%\setcitestyle{square}
\usepackage{url,lineno}
\linenumbers


% BELOW TAKEN FROM rticles plos template
%
% amsmath package, useful for mathematical formulas
\usepackage{amsmath}
% amssymb package, useful for mathematical symbols
\usepackage{amssymb}

% hyperref package, useful for hyperlinks
\usepackage{hyperref}

% graphicx package, useful for including eps and pdf graphics
% include graphics with the command \includegraphics
\usepackage{graphicx}

% Sweave(-like)
\usepackage{fancyvrb}
\DefineVerbatimEnvironment{Sinput}{Verbatim}{fontshape=sl}
\DefineVerbatimEnvironment{Soutput}{Verbatim}{}
\DefineVerbatimEnvironment{Scode}{Verbatim}{fontshape=sl}
\newenvironment{Schunk}{}{}
\DefineVerbatimEnvironment{Code}{Verbatim}{}
\DefineVerbatimEnvironment{CodeInput}{Verbatim}{fontshape=sl}
\DefineVerbatimEnvironment{CodeOutput}{Verbatim}{}
\newenvironment{CodeChunk}{}{}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

\usepackage{color}

% Below is from frontiers
%
\bibliographystyle{frontiersinSCNS}
% Use doublespacing - comment out for single spacing
%\usepackage{setspace}
%\doublespacing


% Leave a blank line between paragraphs instead of using \\



\def\keyFont{\fontsize{8}{11}\helveticabold }

%% ** EDIT HERE **
%% PLEASE INCLUDE ALL MACROS BELOW

%% END MACROS SECTION



  \def\Authors{
  Zhian N. Kamvar\,\textsuperscript{1},
  Jonah C. Brooks\,\textsuperscript{2},
  Niklaus J. Grunwald\,\textsuperscript{1,3*}}
% \\
\def\Address{
  \textsuperscript{1} Botany and Plant Pathology, Oregon State University,  Corvallis,  OR,  USA\\
  \textsuperscript{2} College of Electrical Engineering and Computer Science, Oregon State University,  Corvallis,  OR,  USA\\
  \textsuperscript{3} Horticultural Crops Research Laboratory, USDA-Agricultural Research Service,  Corvallis,  OR,  USA\\
}

  
  \def\firstAuthorLast{Kamvar {et~al.}}
  
  
  \def\corrAuthor{Niklaus J. Grunwald}\def\corrAddress{Horticultural Crops Research Laboratory USDA ARS\\3420 NW Orchard Ave.\\Corvallis, OR, 97330}\def\corrEmail{\href{mailto:grunwaln@science.oregonstate.edu}{\nolinkurl{grunwaln@science.oregonstate.edu}}}
  


\begin{document}
\onecolumn
\firstpage{1}

\title[Efficient clones]{Efficient tools for analyzing genome-wide data of clonal organisms}
\author[\firstAuthorLast]{\Authors}
\address{}
\correspondance{}
\extraAuth{}% If there are more than 1 corresponding author, comment this line and uncomment the next one.
%\extraAuth{corresponding Author2 \\ Laboratory X2, Institute X2, Department X2, Organization X2, Street X2, City X2 , State XX2 (only USA, Canada and Australia), Zip Code2, X2 Country X2, email2@uni2.edu}
\topic{}% If your article is part of a Research Topic, please indicate here which.
\maketitle

\begin{abstract}

Understanding the dynamics of plant pathogen populations is imperative to
understanding plant microbe interactions, adaptation of pathogens to hosts,
reactive breeding and deployment of R genes and design of informed management
strategies. With the advent of high throughput sequencing technologies,
obtaining genomic sequences for representative populations has been easier than
ever before. This in combination with a move towards more reproducible research
has led researchers to move away from traditional stand alone programs for
population genetic analysis to modular programs such as R, which has a vibrant
community of contributers submitting packages tailored for several analyses,
including population genetics. The R package poppr was released to specifically
address issues surrounding analysis of clonal populations and has recently
implemented several tools that better facilitate the analysis of clonal
populations across different levels of hierarchies, implementing multilocus
genotype definitions and the index of association for reduced representation
genomic data, and modular bootstrapping of any genetic distance.

\tiny
 \keyFont{ \section{Keywords:} clonality, population genetics, bootstrap, open source}
\end{abstract}

\section*{Introduction}\label{introduction}
\addcontentsline{toc}{section}{Introduction}

\begin{itemize}
\itemsep1pt\parskip0pt\parsep0pt
\item
  Rise in popularity of reproducible methods \textbar{} Should this go
  in discussion? It is one great advantage of using R!

  \begin{itemize}
  \itemsep1pt\parskip0pt\parsep0pt
  \item
    shitty standalone programs
  \end{itemize}
\end{itemize}

Most computational tools for population genetics are based on concepts
developed for sexual model organisms. Populations that reproduce
clonally or are polyploid are thus difficult to characterize using
classical population genetic tools.

The research community using the R statistical and computing language
(RCoreTeam, 2013) has developed a plethora of new resources for
population genetic analysis (Paradis, 2010; Jombart, 2008). Recently, we
introduced the R package \emph{poppr} specifically developed for
analysis of clonal populations (Kamvar et al., 2014). \emph{Poppr}
previously introduced several novel features including the ability to
conduct a hierarchical analysis across unlimited hierarchies, test for
linkage association, graph minimum spanning networks or provide
bootstrap support for Bruvo's distance in resulting trees. In its first
iteration, \emph{poppr} was appropriate for traditional markers systems,
but not ideally suited to population genomic data resulting from high
throughput sequencing methods as increases in the number of markers
decreased the ability to distinguish clonal genotypes due to higher
chance to detect somatic mutations or genotyping error.

\begin{itemize}
\itemsep1pt\parskip0pt\parsep0pt
\item
  Packages available in R

  \begin{itemize}
  \itemsep1pt\parskip0pt\parsep0pt
  \item
    Previous lack of tools for clonal populations, still lack of tools
    for genomic data sets

    \begin{itemize}
    \itemsep1pt\parskip0pt\parsep0pt
    \item
      How do you deal with clones?
    \item
      How do you detect linkage?
    \item
      How do you know if you've sampled enough?
    \end{itemize}
  \end{itemize}
\end{itemize}

Multiple references get a semi-colon (Wilson et al., 2014, 2014).
References that are simply mentioned, such as Wilson et al. (2014), are
cited without square braces.

\section*{Materials and Methods}\label{materials-and-methods}
\addcontentsline{toc}{section}{Materials and Methods}

In \emph{poppr} version 1.1, a new S4 object was defined to expand the
genind object of \emph{adegenet}. The genclone object formalized the
definitions of multilocus genotypes and population hierarchies by adding
two slots called mlg and hierarchy that carried a numeric vector and a
data frame, respectively. These new slots allow for increased efficiency
and ease of use by allowing these metadata to travel with the genetic
data.

\subsection*{Clone correction}\label{clone-correction}
\addcontentsline{toc}{subsection}{Clone correction}

As highlighted in previous work, clone correction is an important part
of population genetic analysis of organisms that have cryptic growth or
are known to produce asexual propagules (Kamvar et al., 2014; Milgroom,
1996; Grünwald et al., 2003). This method removes bias that would
otherwise affect metrics that rely on allele frequencies. It was
initially designed for data with only a handful of markers. With the
advent of large-scale sequencing and reduced- representation libraries,
it has become easier to sequence tens of thousands of markers from
hundreds of individuals (Elshire et al., 2011; Davey et al., 2011; Davey
and Blaxter, 2010). With this number of markers, the genetic resolution
is much greater, but the chance of genotyping error is also greatly
increased (Mastretta-Yanes et al., 2014). Taking this and somatic
mutation into account, it would be impossible to separate true clones
from independent individuals by just comparing what multilocus genotypes
are different. We introduce a a new transparent method for collapsing
unique multilocus genotypes determined by naive string comparison into
multilocus lineages utilizing any genetic distance given three different
clustering algorithms: farthest neighbor, nearest neighbor, and UPGMA
(average neighbor) {[}citation needed{]}.

The clustering algorithms act on a distance matrix that is either
provided by the user or generated via a function that will calculate a
distance from genclone objects such as \texttt{bruvo.dist}, which will
calculate Bruvo's distance at any level of ploidy. All algorithms have
been implemented in C and utilize the open MP framework for optional
parallel processing. Default is the conservative farthest neighbor
algorithm, which will only cluster samples together only if all samples
in the cluster are at a distance less than the given threshold. By
contrast, the nearest neighbor algorithm will have a chaining effect
that will cluster samples akin to adding links on a chain where a sample
can be included in a cluster if all of the samples have at least one
connection below a given threshold. The UPGMA, or average neighbor
clustering algorithm is the one most familiar to biologists as it is
often used to generate preliminary ultra-metric trees based off of
genetic distance. This algorithm will cluster by creating a
representative sample per cluster and joining clusters if these
representative samples are closer than the given threshold.

\subsubsection*{\texorpdfstring{\emph{P.
infestans}}{P. infestans}}\label{p.-infestans}
\addcontentsline{toc}{subsubsection}{\emph{P. infestans}}

\emph{Phytophthora infestans} is the causal agent of potato late blight
originating from Mexico and spread to Europe in the mid 19th century
{[}Goss et al. (2014); cooke{]}. \emph{P. infestans} reproduces both
clonally and sexually. The clonal lineages of \emph{P. infestans} have
been formally defined into 18 separate clonal lineages using various
molecular methods. We utilize these data to show how the
\texttt{mlg.filter} function collapses multilocus genotypes with Bruvo's
distance assuming a genome addition model (Bruvo et al., 2004).

\subsection*{Index of association}\label{index-of-association}
\addcontentsline{toc}{subsection}{Index of association}

The index of association (\(I_A\)) is a measure of multilocus linkage
disequilibrium that is most often used to detect clonal reproduction
within organisms that have the ability to reproduce via sexual or
asexual processes (brown1980; maynard1993; Milgroom, 1996). It was
standardized in 2001 as \(\bar{r}_d\) by (Agapow and Burt, 2001) to
address the issue of scaling with increasing number of loci. This metric
is typically applied to traditional dominant and co-dominant markers
such as AFLPs, SNPs, or microsatellite markers. With the advent of high
throughput sequencing, SNP data is now available in in a genome-wide
context and in very large matrices including thousands of SNPs. Thus,
the likelihood of finding mutations within two individuals of a given
clone increases and tools are needed for defining clone boundaries. For
this reason, we devised two approaches using the index of association
for large numbers of markers typical for population genomic studies.

The first approach is a sliding window approach that would utilize the
position of markers in the genome to calculate \(\bar{r}_d\) among any
number of SNPs found within the windowed region. It is important that
this calculation utilize \(\bar{r}_d\) as the number of loci will be
different within each window (Agapow and Burt, 2001). This approach
would be suited for a quick calculation of linkage disequilibrium across
the genome that can detect potential hotspots of LD that could be
investigated further with more computationally intensive methods
assuming that the number of samples \textless{}\textless{} the number of
loci.

A sliding window approach would not be good for utilizing \(\bar{r}_d\)
as a test for clonal reproduction as it would necessarily focus on loci
within a short section of the genome that may or may not be recombining.
A remedy for this is to randomly sample \(m\) loci, calculate
\(\bar{r}_d\) and repeat \(r\) times, creating a distribution of
expected values of \(\bar{r}_d\).

As the data for this would be stored in a genlight object, which stores
genetic information in bits, the calculations of \(\bar{r}_d\) are
performed in parallelized C code for efficiency (Jombart, 2008).

\begin{itemize}
\itemsep1pt\parskip0pt\parsep0pt
\item
  Clone correction based on genetic distance

  \begin{itemize}
  \itemsep1pt\parskip0pt\parsep0pt
  \item
    reduced representation data (GBS)
  \item
    consistent clones with one or two highly variable loci (Cooke
    example)
  \item
    nearest neighbor; farthest neighbor; average neighbor
  \end{itemize}
\item
  Index of Association

  \begin{itemize}
  \itemsep1pt\parskip0pt\parsep0pt
  \item
    Invented for small number of markers
  \item
    Methods of \(\bar{r}_d\) for large number of markers:
  \item
    sliding window (could be a fast way to detect linkage)
  \item
    random sampling of x loci (creates a distribution of \(\bar{r}_d\))
  \end{itemize}
\item
  Bootstrapping genetic distance

  \begin{itemize}
  \itemsep1pt\parskip0pt\parsep0pt
  \item
    Matrix algebra makes it easy to calculate genetic distance in R
    (with alleles in columns).
  \item
    Alleles in columns makes it hard to bootstrap (alleles are not
    independent units).
  \end{itemize}
\item
  Population Stratification

  \begin{itemize}
  \itemsep1pt\parskip0pt\parsep0pt
  \item
    Necessary for analyses of factorial sampling (Everhart and Scherm,
    2014) or hierarchical sampling (Linde et al., 2002)
  \item
    Implementation in StrataG and Hierfstat loosely defined.
  \item
    Solution in S4 classes: strict definition.
  \item
    adopted by larger community
  \end{itemize}
\end{itemize}

\subsection*{Additions}\label{additions}
\addcontentsline{toc}{subsection}{Additions}

\begin{itemize}
\itemsep1pt\parskip0pt\parsep0pt
\item
  Psex
\item
  Bootstrapping for individuals or populations for any genetic distance
\item
  MLG clustering
\item
  Index of Association for genomic sequences
\item
  Parallelization
\item
  Genotype Accumulation Curve
\item
  Matching Ties in Minimum Spanning networks (Cite ramorum paper)
\end{itemize}

\section*{Results}\label{results}
\addcontentsline{toc}{section}{Results}

\subsection*{Clonal identification}\label{clonal-identification}
\addcontentsline{toc}{subsection}{Clonal identification}

\subsubsection*{\texorpdfstring{\emph{P.
infestans}}{P. infestans}}\label{p.-infestans-1}
\addcontentsline{toc}{subsubsection}{\emph{P. infestans}}

The three algorithms were able to detect 18 multilocus lineages at
different distances (Fig. 1). Contingency tables between the described
multilocus genotypes and the genotypes defined by distance show that
most of the 18 lineages were resolved, except for US-8, which is
polytomic (Table 1).

\begin{CodeChunk}

\includegraphics{poppr_frontiers_files/figure-latex/pinf_data-1} \end{CodeChunk}\begin{CodeChunk}

\includegraphics{poppr_frontiers_files/figure-latex/pinf_tree-1} \end{CodeChunk}\begin{table}[ht]
\centering
\begin{tabular}{ccccccccccccccccccc}
  \hline
 & 1 & 3 & 5 & 8 & 9 & 10 & 12 & 13 & 15 & 17 & 19 & 20 & 23 & 24 & 25 & 27 & 28 & 29 \\ 
  \hline
B & . & . & . & . & . & . & . & . & 1 & . & . & . & . & . & . & . & . & . \\ 
  C & . & 1 & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . \\ 
  EU-13 & . & . & . & . & 1 & . & . & . & . & . & . & . & . & . & . & . & . & . \\ 
  EU-4 & . & . & . & . & . & 1 & . & . & . & . & . & . & . & . & . & . & . & . \\ 
  EU-8 & . & . & . & . & . & . & . & 1 & . & . & . & . & . & . & . & . & . & . \\ 
  EU-5 & . & . & . & . & . & . & 2 & . & . & . & . & . & . & . & . & . & . & . \\ 
  US-8 & . & . & . & . & . & . & . & . & . & . & . & 1 & . & 2 & 1 & . & . & . \\ 
  D.1 & 1 & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . \\ 
  D.2 & 1 & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . \\ 
  US-11 & . & . & . & . & . & . & . & . & . & 2 & . & . & . & . & . & . & . & . \\ 
  US-12 & . & . & 1 & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . \\ 
  US-14 & . & . & . & . & . & . & . & . & . & . & . & . & . & 1 & . & . & . & . \\ 
  US-17 & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & 1 & . \\ 
  US-20 & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & 2 & . & . \\ 
  US-21 & . & . & . & . & . & . & . & . & . & . & 2 & . & . & . & . & . & . & . \\ 
  US-22 & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & 2 \\ 
  US-23 & . & . & . & 3 & . & . & . & . & . & . & . & . & . & . & . & . & . & . \\ 
  US-24 & . & . & . & . & . & . & . & . & . & . & . & . & 3 & . & . & . & . & . \\ 
   \hline
\end{tabular}
\end{table}

\subsection*{Speed}\label{speed}
\addcontentsline{toc}{subsection}{Speed}

\subsection*{Community Acceptance}\label{community-acceptance}
\addcontentsline{toc}{subsection}{Community Acceptance}

\begin{itemize}
\itemsep1pt\parskip0pt\parsep0pt
\item
  articles that have cited poppr
\item
  hierarchical methods have been ported to adegenet
\end{itemize}

\section*{Discussion}\label{discussion}
\addcontentsline{toc}{section}{Discussion}

\begin{itemize}
\itemsep1pt\parskip0pt\parsep0pt
\item
  loops in graphs allow for analysis of population structure via graph
  walking algorithms in igraph
\item
  bootstrapping methods encourage future developers to write distance
  implementations in common format
\item
  moving towards open source, modular tools is the direction that
  population genetics and plant pathology needs to go.
\end{itemize}

\section*{References}\label{references}
\addcontentsline{toc}{section}{References}

Agapow, P.-M., and Burt, A. (2001). Indices of multilocus linkage
disequilibrium. \emph{Molecular Ecology Notes} 1, 101--102.
doi:\href{http://dx.doi.org/10.1046/j.1471-8278.2000.00014.x}{10.1046/j.1471-8278.2000.00014.x}.

Bruvo, R., Michiels, N. K., D'Souza, T. G., and Schulenburg, H. (2004).
A simple method for the calculation of microsatellite genotype distances
irrespective of ploidy level. \emph{Molecular Ecology} 13, 2101--2106.

Davey, J. W., and Blaxter, M. L. (2010). RADSeq: Next-generation
population genetics. \emph{Briefings in Functional Genomics} 9,
416--423.
doi:\href{http://dx.doi.org/10.1093/bfgp/elq031}{10.1093/bfgp/elq031}.

Davey, J. W., Hohenlohe, P. A., Etter, P. D., Boone, J. Q., Catchen, J.
M., and Blaxter, M. L. (2011). Genome-wide genetic marker discovery and
genotyping using next-generation sequencing. \emph{Nature Reviews
Genetics} 12, 499--510.

Elshire, R. J., Glaubitz, J. C., Sun, Q., Poland, J. A., Kawamoto, K.,
Buckler, E. S., and Mitchell, S. E. (2011). A robust, simple
genotyping-by-sequencing (gBS) approach for high diversity species.
\emph{PloS one} 6, e19379.

Everhart, S., and Scherm, H. (2014). Fine-scale genetic structure of
monilinia fructicola during brown rot epidemics within individual peach
tree canopies. \emph{Phytopathology}.

Goss, E. M., Tabima, J. F., Cooke, D. E., Restrepo, S., Fry, W. E.,
Forbes, G. A., Fieland, V. J., Cardenas, M., and Gr{ü}nwald, N. J.
(2014). The irish potato famine pathogen phytophthora infestans
originated in central mexico rather than the andes. \emph{Proceedings of
the National Academy of Sciences} 111, 8791--8796.

Grünwald, N. J., Goodwin, S. B., Milgroom, M. G., and Fry, W. E. (2003).
Analysis of genotypic diversity data for populations of microorganisms.
\emph{Phytopathology} 93, 738--46. Available at:
\url{http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO.2003.93.6.738}.

Jombart, T. (2008). Adegenet: a R package for the multivariate analysis
of genetic markers. \emph{Bioinformatics} 24, 1403--1405.
doi:\href{http://dx.doi.org/10.1093/bioinformatics/btn129}{10.1093/bioinformatics/btn129}.

Kamvar, Z. N., Tabima, J. F., and Gr{ü}nwald, N. J. (2014). Poppr: An r
package for genetic analysis of populations with clonal, partially
clonal, and/or sexual reproduction. \emph{PeerJ} 2, e281.

Linde, C., Zhan, J., and McDonald, B. (2002). Population structure of
mycosphaerella graminicola: From lesions to continents.
\emph{Phytopathology} 92, 946--955.

Mastretta-Yanes, A., Zamudio, S., Jorgensen, T. H., Arrigo, N., Alvarez,
N., Pi{ñ}ero, D., and Emerson, B. C. (2014). Gene duplication,
population genomics, and species-level differentiation within a tropical
mountain shrub. \emph{Genome biology and evolution} 6, 2611--2624.

Milgroom, M. G. (1996). Recombination and the multilocus structure of
fungal populations. \emph{Annual Review of Phytopathology} 34, 457--477.

Paradis, E. (2010). Pegas: an R package for population genetics with an
integrated--modular approach. \emph{Bioinformatics} 26, 419--420.

RCoreTeam (2013). R: A Language and Environment for Statistical
Computing. Available at: \url{http://www.R-project.org/}.

Wilson, G., Aruliah, D., Brown, C. T., Hong, N. P. C., Davis, M., Guy,
R. T., Haddock, S. H., Huff, K. D., Mitchell, I. M., Plumbley, M. D., et
al. (2014). Best practices for scientific computing. \emph{PLoS biology}
12, e1001745.


\end{document}

