---
title: "Reproducible research for clonal organisms"
runtitle: "Reproducible clones"
author:
  - name: Zhian N. Kamvar
    affiliation: '1'
    etal: Kamvar
  - name: Jonah C. Brooks
    affiliation: '2'
  - name: Niklaus J. Grunwald
    affiliation: '1,3'
    email: grunwaln@science.oregonstate.edu
    institution: Horticultural Crops Research Laboratory USDA ARS
    street: 3420 NW Orchard Ave.
    city: Corvallis
    state: OR
    zip: 97330
affiliation:
  - id: '1'
    department: Botany and Plant Pathology
    institution: Oregon State University
    city: Corvallis
    state: OR
    country: USA
  - id: '2' 
    department: College of Electrical Engineering and Computer Science
    institution: Oregon State University
    city: Corvallis
    state: OR
    country: USA
  - id: '3' 
    department: Horticultural Crops Research Laboratory
    institution: USDA-Agricultural Research Service
    city: Corvallis
    state: OR
    country: USA
output: FrontiersTemplate::frontiers_article
csl: frontiers.csl
bibliography: citations.bib
---


\begin{abstract}

Understanding the dynamics of plant pathogen populations is imperative to
understanding what management strategies are important. With the advent of more
advanced and cheaper sequencing technologies, obtaining genomic sequences for
representative populations has been easier than ever before. This in combination
with a move towards more reproducible research has lead researchers to move away
from traditional stand alone programs for population genetic analysis to modular
programs such as R, which has a vibrant community of contributers submitting
packages tailored for several analyses, including population genetics. The R
package poppr was released to specifically address issues surrounding analysis
of clonal populations and has recently implemented several tools that better
facilitate the analysis of clonal populations across different levels of
hierarchies, implementing multilocus genotype definitions and the index of
association for reduced representation genomic data, and modular bootstrapping
of any genetic distance.

\tiny
 \keyFont{ \section{Keywords:} clonality population genetics bootstrap open source}
\end{abstract}


# Introduction {-}

Multiple references get a semi-colon [@wilson2014best; @wilson2014best]. References that
are simply mentioned, such as @wilson2014best, are cited without square braces.

- Rise in popularity of reproducible methods
    - shitty standalone programs 
- Packages available in R
    - Previous lack of tools for clonal populations, still lack of tools for genomic data sets
        - How do you deal with clones?
        - How do you detect linkage? 
        - How do you know if you've sampled enough?

# Materials and Methods {-}

## Clone correction {-}

As highlighted in previous work, clone correction is an important part of
population genetic analysis of organisms that have cryptic growth or are known
to produce asexual propagules [@kamvar2014poppr; @milgroom1996recombination;
@grunwald2003analysis]. This method removes bias that would otherwise affect
metrics that rely on allele frequencies. It was initially designed for data with
only a handful of markers. With the advent of large-scale sequencing and
reduced- representation libraries, it has become easier to sequence tens of
thousands of markers from hundreds of individuals [@elshire2011robust;
@davey2011genome; @davey2010rad]. With this number of markers, the genetic
resolution is much greater, but the chance of genotyping error is also greatly
increased [@mastretta2014gene]. Taking this and somatic mutation into account,
it would be impossible to separate true clones from independent individuals by
just comparing what multilocus genotypes are different. We provide a transparent
method for collapsing unique multilocus genotypes determined by naive string
comparison into multilocus lineages utilizing any genetic distance given three
different clustering algorithms: farthest neighbor, nearest neighbor, and
average neighbor [citation needed]. 

## Index of association {-}

The index of association ($I_A$)is a measure of multilocus linkage
disequilibrium that is most often used to detect clonal reproduction within
organisms that have the ability to reproduce via sexual or asexual processes
[brown1980; maynard1993; @milgroom1996recombination]. It was standardized in
2001 as $\bar{r}_d$ by agapow 2001 to address the issue of scaling with
increasing number of loci. When this metric came out, it was utilized for
dominant and co-dominant markers such as AFLPs, SNP panels, and microsatellite
markers. Unfortunately, it would be erroneous and wasteful to apply the index of
association to large numbers of markers as chances of linkage are expected to go
up, weakening the any true signal of clonal reproduction present in the data.
For this reason, we have devised two approaches for addressing using the index
of association for large numbers of markers
 
 - Clone correction based on genetic distance
    - reduced representation data (GBS)
    - consistent clones with one or two highly variable loci (Cooke example)
    - nearest neighbor; farthest neighbor; average neighbor
 - Index of Association
    - Invented for small number of markers
    - Methods of $\bar{r}_d$ for large number of markers:
      - sliding window (could be a fast way to detect linkage)
      - random sampling of x loci (creates a distribution of $\bar{r}_d$)
 - Bootstrapping genetic distance
    - Matrix algebra makes it easy to calculate genetic distance in R (with alleles in columns).
    - Alleles in columns makes it hard to bootstrap (alleles are not independent units).
 - Population Stratification
    - Necessary for analyses of factorial sampling [@everhart2014fine] or hierarchical sampling [@linde2002population]
    - Implementation in StrataG and Hierfstat loosely defined.
    - Solution in S4 classes: strict definition.
    - adopted by larger community
    

## Additions {-}

- Psex
- Bootstrapping for individuals or populations for any genetic distance
- MLG clustering
- Index of Association for genomic sequences
- Parallelization
- Genotype Accumulation Curve
- Matching Ties in Minimum Spanning networks (Cite ramorum paper)

# Results {-}

## Speed {-}

## Community Acceptance {-}

- articles that have cited poppr
- hierarchical methods have been ported to adegenet

# Discussion {-}

- loops in graphs allow for analysis of population structure via graph walking algorithms in igraph
- bootstrapping methods encourage future developers to write distance implementations in common format
- moving towards open source, modular tools is the direction that population genetics and plant pathology needs to go. 




# References {-}