---
title: "Novel tools for analyzing genome-wide data of clonal populations"
runtitle: "Novel tools for population genetics"
author:
  - name: Zhian N. Kamvar
    affiliation: '1'
    etal: Kamvar
  - name: Jonah C. Brooks
    affiliation: '2'
  - name: Niklaus J. Grunwald
    affiliation: '1,3'
    email: grunwaln@science.oregonstate.edu
    institution: Horticultural Crops Research Laboratory USDA ARS
    street: 3420 NW Orchard Ave.
    city: Corvallis
    state: OR
    zip: 97330
affiliation:
  - id: '1'
    department: Botany and Plant Pathology
    institution: Oregon State University
    city: Corvallis
    state: OR
    country: USA
  - id: '2' 
    department: College of Electrical Engineering and Computer Science
    institution: Oregon State University
    city: Corvallis
    state: OR
    country: USA
  - id: '3' 
    department: Horticultural Crops Research Laboratory
    institution: USDA-Agricultural Research Service
    city: Corvallis
    state: OR
    country: USA
output: FrontiersTemplate::frontiers_article
csl: frontiers.csl
bibliography: citations.bib
---

<!--
Note: Everything in the abstract needs to be in latex format.
-->

\begin{abstract}

To gain a detailed understanding of plant microbe interactions, adaptation of
pathogens to hosts, develop reactive plant breeding programs, and to deploy R
genes effectively, knowledge of the population dynamics and evolutionary history
of populations is crucial. With the advent of high throughput sequencing
technologies, obtaining genomic sequences for representative populations has
become easier than ever before. A move towards open, reproducible science has
provided impetus for developing population genetic analysis tools in R. We
previously contributed the R package \textit{poppr} specifically addressing
issues with analysis of clonal populations. In this paper we provide several
significant extensions to \textit{poppr} with a focus on large, genome wide SNP
data. Specifically, we provide analyses across any level of hierarchies, a new
function to define clone boundaries we call \texttt{clone\_caller} allowing for
inspection and definition of what is a clonal lineage, and the index of
association for reduced representation genomic data, and modular bootstrapping
of any genetic distance.

\tiny
 \keyFont{ \section{Keywords:} clonality, population genetics, bootstrap, open source}
\end{abstract}


# Introduction {-}

To paraphrase Dobzhansky, nothing in the field of plant-microbe interactions
makes sense except in the light of population genetics [@dobzhansky2013nothing].
Genetic forces such as selection and drift act on alleles in a population. Thus,
a true understanding of how plant pathogens evolve and adapt to crops,
fungicides, or other factors, can only emerge in the context of population level
phenomena given the demographic history of populations [@Mcdonald2002;
@grunwald2011evolution; @milgroom1989population]. The field of population
genetics, in the era of whole genome resequencing, provides unprecedented power
to describe the evolutionary history and population processes that drive
coevolution between pathogens and hosts. This powerful field thus critically
enables effective deployment of R genes, design of pathogen informed plant
resistance breeding programs, and implementation of fungicide rotations that
minimize emergence of resistance.

Most computational tools for population genetics are based on concepts developed
for sexual model organisms. Populations that reproduce clonally or are polyploid
are thus difficult to characterize using classical population genetic tools
because theoretical assumptions underlying the theory are violated. Yet, many
plant pathogen populations are at least partially clonal if not completely
clonal [@milgroom1996recombination; @anderson1995clonality].  Thus, development
of tools for analysis of clonal or polyploid populations is needed.

Genotyping by sequencing and whole genome resequencing provide the unprecedented
ability to identify >1,000 single nucleotide polymorphisms (SNPs) in populations
[@elshire2011robust; @luikart2003power; @davey2011genome]. Availability of these
large SNP data sets provides new challenges for data analysis. For example, it
is not clear what a clone is in large SNP data where the chance of observing
variation at a given SNP locus within independent samples of the same clone are
substantial enough that novel tools for definition of clone boundaries are
required. With traditional marker data (e.g., SSR, AFLP) a clone was typically
defined as a unique multilocus genotype (MLG). However, with large SNP data a
measure of genetic distance is required to define the boundary of an MLG (e.g.,
clone) or the boundaries of a clonal lineage. Definition of a clone is further
complicated by the presence of missing data that is typical for reduced
representation libraries used in GBS or genome re-sequencing. Are two
individuals that are identical for all observed SNPs different if one additional
locus has missing data?

<!-- This doesn't flow right -->
Another major issues with large SNP data is computational efficiency. Analyses
that resample a data set or rely on MCMC are computationally slow for large data
and thus are preferentially implemented in a low level language such as C rather
than in interpreted languages such as R.

The research community using the R statistical and computing language [@R] has
developed a plethora of new resources for population genetic analysis [@pegas;
@Jombart_2008]. Recently, we introduced the R package *poppr* specifically
developed for analysis of clonal populations [@kamvar2014poppr]. *Poppr*
previously introduced several novel features including the ability to conduct a
hierarchical analysis across unlimited hierarchies, test for linkage
association, graph minimum spanning networks or provide bootstrap support for
Bruvo's distance in resulting trees. It was well received by the community,
garnering 14 citations in its first year of publication.

In version 1.1, to address difficulties with handling hierarchical and
mutlilocus genotypic metadata, a new S4 object called "genclone" was defined to
expand the genind object of *adegenet*. The genclone object formalized the
definitions of multilocus genotypes and population hierarchies by adding two
slots called mlg and hierarchy that carried a numeric vector and a data frame,
respectively. These new slots allow for increased efficiency and ease of use by
allowing these metadata to travel with the genetic data. The addition of the
population hierarchies has proved to be advantageous enough that they have
recently been adopted into the more central adegenet package [@Jombart_2008].

In version 1, *poppr* was appropriate for traditional markers systems, but not
well suited to population genomic data resulting from high throughput sequencing
methods. The raw size of these data made it difficult to conduct traditional
analyses. Here, we introduce *poppr* 2.0, which provides a significant update to
*poppr* including novel tools for analysis of clonal populations specifically
for large SNP data. Significant novel tools include functions for calculating
clone boundaries and collapsing individuals into user-specified clones based on
genetic distance, sliding window analyses, genotype accumulation curves,
reticulations in minimum spanning networks, and bootstrapping for any genetic
distance.

<!--

Multiple references get a semi-colon [@wilson2014best; @wilson2014best].
References that are simply mentioned, such as @wilson2014best, are cited without
square braces.

-->

# Materials and Methods {-}

## Clonal identification {-}

As highlighted in previous work, clone correction is an important part of
population genetic analysis of organisms that have cryptic growth or are known
to produce asexual propagules [@kamvar2014poppr; @milgroom1996recombination;
@grunwald2003analysis]. This method removes bias that would otherwise affect
metrics that rely on allele frequencies. It was initially designed for data with
only a handful of markers. With the advent of large-scale sequencing and
reduced- representation libraries, it has become easier to sequence tens of
thousands of markers from hundreds of individuals [@elshire2011robust;
@davey2011genome; @davey2010rad]. With this number of markers, the genetic
resolution is much greater, but the chance of genotyping error is also greatly
increased [@mastretta2015restriction]. Taking this and somatic mutation into
account, it would be impossible to separate true clones from independent
individuals by just comparing what multilocus genotypes are different. We
introduce a a new transparent method for collapsing unique multilocus genotypes
determined by naive string comparison into multilocus lineages utilizing any
genetic distance given three different clustering algorithms: farthest neighbor,
nearest neighbor, and UPGMA (average neighbor) [@sokal1958statistical].

The clustering algorithms act on a distance matrix that is either provided by
the user or generated via a function that will calculate a distance from
genclone objects such as `bruvo.dist`, which will calculate Bruvo's distance at
any level of ploidy [@bruvo2004simple]. All algorithms have been implemented in
C and utilize the open MP framework for optional parallel processing. Default is
the conservative farthest neighbor algorithm, which will only cluster samples
together only if all samples in the cluster are at a distance less than the
given threshold. By contrast, the nearest neighbor algorithm will have a
chaining effect that will cluster samples akin to adding links on a chain where
a sample can be included in a cluster if all of the samples have at least one
connection below a given threshold. The UPGMA, or average neighbor clustering
algorithm is the one most familiar to biologists as it is often used to generate
preliminary ultra-metric trees based off of genetic distance. This algorithm
will cluster by creating a representative sample per cluster and joining
clusters if these representative samples are closer than the given threshold.

## Demonstration data: *P. infestans* {-}

We utilize these data from *Phytophthora infestans* to show how the `mlg.filter`
function collapses multilocus genotypes with Bruvo's distance assuming a genome
addition model [@bruvo2004simple]. *P. infestans* is the causal agent of potato
late blight originating from Mexico and spread to Europe in the mid 19th century
[@goss2014irish; @li2013efficient; @lees2006novel]. *P. infestans* reproduces
both clonally and sexually. The clonal lineages of *P. infestans* have been
formally defined into 18 separate clonal lineages using a combination of various
molecular methods including AFLP and microsatellite markers [@lees2006novel].
For these data, we used `mlg.filter` to detect all of the distance thresholds at
which 18 multilocus lineages would be resolved. We the used these thresholds to
define multilocus lineages and create contingency tables and dendrograms to
determine how well the multilocus lineages were detected.

## Simulated data {-}

To detect clonal lineages utilizing genomic data, we utilized simulated data
constructed using the `glSim` function in adegenet [@jombart2011adegenet]. For
the simulations, two diploid data sets were created, each with 10k SNPs (25%
structured into two groups) and 200 samples with 10 ancestral populations of
even sizes. Clones were created in one data set by marking each sample with a
unique identifier and then randomly sampling with replacement. It is well
documented that reduced- representation sequencing can introduce several
erroneous calls and missing data [@mastretta2015restriction]. To reflect this,
we mutated SNPs at a rate of 10% and inserted an average of 10% missing data for
each sample after clones were created, ensuring that no two sequences were
alike. The number of mutations and missing data per sample was determined with a
sampling from a poisson distribution with $\lambda = 1000$. After pooling, 20%
of the data set was randomly sampled for analysis. Genetic distance was
represented as the fraction of different sites between samples, counting missing
data as equivalent in comparison. Since each data set would have varying
distances between samples, the clonal boundary threshold was defined as the
midpoint of the largest gap between two thresholds that collapsed less than 50%
of the data.

## Index of association {-}

The index of association ($I_A$) is a measure of multilocus linkage
disequilibrium that is most often used to detect clonal reproduction within
organisms that have the ability to reproduce via sexual or asexual processes
[@brown1980multilocus; @smith1993how; @milgroom1996recombination]. It was
standardized in 2001 as $\bar{r}_d$ by @Agapow_2001 to address the issue of
scaling with increasing number of loci. This metric is typically applied to
traditional dominant and co-dominant markers such as AFLPs, SNPs, or
microsatellite markers. With the advent of high throughput sequencing, SNP data
is now available in in a genome-wide context and in very large matrices
including thousands of SNPs. Thus, the likelihood of finding mutations within
two individuals of a given clone increases and tools are needed for defining
clone boundaries. For this reason, we  devised two approaches using the index of
association for large numbers of markers typical for population genomic studies.

The first approach is a sliding window approach that would utilize the position
of markers in the genome to calculate $\bar{r}_d$ among any number of SNPs found
within the windowed region. It is important that this calculation utilize
$\bar{r}_d$ as the number of loci will be different within each window
[@Agapow_2001]. This approach would be suited for a quick calculation of linkage
disequilibrium across the genome that can detect potential hotspots of LD that
could be investigated further with more computationally intensive methods
assuming that the number of samples << the number of loci.

As it would necessarily focus on loci within a short section of the genome that
may or may not be recombining, a sliding window approach would not be good for
utilizing $\bar{r}_d$ as a test for clonal reproduction. A remedy for this is to
randomly sample $m$ loci, calculate $\bar{r}_d$ and repeat $r$ times, creating a
distribution of expected values of $\bar{r}_d$.

<!--
 - Clone correction based on genetic distance
    - reduced representation data (GBS)
    - consistent clones with one or two highly variable loci (Cooke example)
    - nearest neighbor; farthest neighbor; average neighbor
 - Index of Association
    - Invented for small number of markers
    - Methods of $\bar{r}_d$ for large number of markers:
      - sliding window (could be a fast way to detect linkage)
      - random sampling of x loci (creates a distribution of $\bar{r}_d$)
 - Bootstrapping genetic distance
    - Matrix algebra makes it easy to calculate genetic distance in R (with alleles in columns).
    - Alleles in columns makes it hard to bootstrap (alleles are not independent units).
 - Population Stratification
    - Necessary for analyses of factorial sampling [@everhart2014fine] or hierarchical sampling [@linde2002population]
    - Implementation in StrataG and Hierfstat loosely defined.
    - Solution in S4 classes: strict definition.
    - adopted by larger community
-->

## Population Strata and Hierarchies {-}

Assessments of population structure through methods such as hierarchical
$F_{st}$ and AMOVA benefit greatly from multiple levels of population definition
[@linde2002population; @everhart2014fine; @grunwald2006hierarchical]. With
clonal organisms, basic practice has been to clone-censor data to avoid downward
bias in diversity due to duplicated genotypes that may or may not represent
different samples [@milgroom1996recombination]. Data structures for population
genetic data mostly allow for only one level of hierarchical definition. The
impetus was placed on the researchers to provide the population hierarchies for
every step of the analysis. In poppr version 1.1, the hierarchy slot was
introduced to allow unlimited population hierarchies or stratifications to
travel with the data. In practice, it is stored as a data frame where each
column represents a separate hierarchical level. This is then used to set the
population factor of the data by supplying a hierarchical formula containing one
or more column names of the data frame in the hierarchy slot. This
functionality, developed in *poppr*, has been moved to the *adegenet* package in
version 2.0 and the slot and methods have been renamed to `strata`.

## Genotype Accumulation Curve {-}

Analysis of population genetics of clonal organisms often borrows from
ecological methods such as analysis of diversity within populations
[@milgroom1996recombination; @arnaud2007standardizing; @grunwald2003analysis].
When choosing markers for analysis, it is important to make sure that the
observed diversity in your sample will not appreciably increase if an additional
marker is added [@arnaud2007standardizing]. This concept is analogous to a
species accumulation curve, obtained by rarefaction. The genotype accumulation
curve in *poppr* is implemented by randomly sampling $x$ loci and counting the
number of observed MLGs. This repeated $r$ times for 1 locus up to $n-1$ loci,
creating $n-1$ distributions of observed MLGs.

## Minimum Spanning Network Reticulation {-}

*Poppr* introduced minimum spanning networks in it's original iteration that
were based off of igraph's function `minimum.spanning.tree` [@csardi2006igraph].
This algorithm would produce a minimum spanning tree, but with no reticulations,
which is common and expected of clonal organisms. In other minimum spanning
network programs, reticulation is obtained by calculating the minimum spanning
tree several times and returning the set of all edges included in the trees. Due
to the way igraph has implemented Prim's algorithm, it is not possible to
utilize this strategy, thus we have implemented a C function to walk through a
distance matrix and minimum spanning tree to connect groups of nodes with edges
of equal weight. 

To demonstrate the utility of minimum spanning network reticulation, we used two
clonal data sets: H3N2 flu virus data from the `adegenet` package using years of
each epidemic as the population factor, and *Phytophthora ramorum* data from
Nurseries and Oregon forests [@jombart2010discriminant; @kamvar2014sudden].
Minimum spanning networks were created with and without reticulation using the
*poppr* functions `diss.dist` and `bruvo.msn` for the H3N2 and *P. ramorum*
data, respectively [@kamvar2014poppr; @bruvo2004simple]. To detect mlg clusters,
the infoMAP community detection algorithm was applied with 10,000 trials as
implemented in the R package `igraph` version 0.7.1 utilizing genetic distance
as edge weights and number of samples in each MLG as vertex weights
[@csardi2006igraph; @rosvall2008maps].

To evaluate the results, we compared the number and size of resulting
communities as we expect a highly clonal organism with low genetic diversity to
result in few, large communities. We also created contingency tables of the
community assignments with the defined populations and used those to calculate
entropy using Shannon's index with the function `diversity` from the R package
`vegan` version 2.2-1 [@oksanen2015vegan; @shannon2001mathematical]. A low
entropy indicates few large communities whereas high entropy indicates many
small communities.


## Bootstrapping {-}

Calculating genetic distance for among samples and populations is very important
method for assessing population differentiation through methods such as
$F_{st}$, AMOVA, and Mantel tests. Confidence in distance metrics is related to
the confidence in the markers to accurately represent the diversity of the data.
Especially true with microsatellite markers, a single hyper-diverse locus can
make a population appear to have more diversity based on genetic distance. Using
a bootstrapping procedure of randomly sampling loci with replacement when
calculating a distance matrix gives confidence in hierarchical clustering.
Because genetic data in a genind object is represented as a matrix with samples
in rows and alleles in columns, bootstrapping is a non-trivial task as all
alleles in a single locus need to be sampled together. To remedy this, we have
created an internal S4 class called "bootgen", which extends the internal "gen"
class from adegenet. This class can be created from any genind, genclone, or
genpop object, and allows loci to be sampled with replacement. To further
facilitate bootstrapping, a function called aboot, which stands for "any boot",
is introduced that will bootstrap any genclone, genind, or genpop object with
any genetic distance that can be calculated from it.

# Results {-}

## Clonal identification {-}

With the *P. infestans* data, the three algorithms were able to detect 18
multilocus lineages at different distance thresholds (Fig. 1). Contingency
tables between the described multilocus genotypes and the genotypes defined by
distance show that most of the 18 lineages were resolved, except for US-8, which
is polytomic (Table 1). Out of the 100 simulations run, we found that across all
methods, detection of duplicated samples had $\sim$ 98% true positive fraction
and $\sim$ 0.8% false positive fraction indicating that this method is robust to
simulated populations.


> Nik, these are just to see how the figures look in the paper, the actual figures need to be uploaded separately, according to the frontiers specifications.

```{r, pinf_data, echo = FALSE, message = FALSE}
source("../Rscripts/my_functions.R")
library('poppr')
library('ape')
library('phangorn')
library('animation')
infdat <- RCurl::getURL("https://raw.githubusercontent.com/grunwaldlab/phytophthora_id/master/shiny-server/www/genoid_infestans/reduced_database.txt.csv")
infdat <- read.table(text = infdat, head = TRUE)
pinf   <- df2genind(infdat[-c(1,2)], sep = "/", ploidy = 3, ind.names = infdat[[1]], pop = infdat[[2]])
ssr    <- c(3, 3, 2, 3, 3, 2, 2, 3, 3, 3, 3, 3)
x      <- as.genclone(pinf)
fstats <- filter_stats(x, bruvo.dist, plot = TRUE, replen = ssr, loss = FALSE, nclone = 18)
title(main = expression(paste(italic("P. infestans"), " reference isolates (12 SSR loci)")))
thresh18 <- fstats$average$thresholds[nInd(x) - 18] + .Machine$double.eps^0.5
z <- mlg.filter(x, threshold = thresh18, distance = bruvo.dist, replen = ssr, loss = FALSE, stats = "MLGS")
```
```{r, pinf_tree, echo = FALSE, message = FALSE, fig.height = 10, fig.width = 5}
pinfupgma <- upgma(bruvo.dist(x, replen = ssr, loss = FALSE))
color_mlg_tree(x, pinfupgma, z)
axisPhylo(3)
```
```{r pinf_table, echo = FALSE, message = FALSE, results='asis'}
library("xtable")

res <- table(pop(x), z, dnn = NULL)
res[res == 0] <- "."
print.xtable(xtable(res, align = rep("c", ncol(res) + 1)), comment = FALSE)
```


## Minimum Spanning Networks {-}


The infoMAP algorithm revealed 63 communities with a maximum community size of
77 and $H = 3.56$ for the reticulate network of the H3N2 data and 117
communities with a maximum community size of 26 and $H = 4.65$ for the minimum
spanning tree. The entropy across years was greatly decreased for all
populations with the reticulate network compared to the minimum spanning tree
(Fig 2).

![Fig. 2](poppr_frontiers_files/custom_figures/Figure-2.png)

Graph walking of the reticulated minimum spanning network of *P. ramorum* by the
infoMAP algorithm revealed 16 communities with a maximum community size of 13
and $H = 2.60$. The un-reticulated minimum spanning tree revealed 20 communities
with a maximum community size of 7 and $H = 2.96$. In the ability to predict
Hunter Creek as belonging to a single community, the reticulated network was
successful whereas the minimum spanning tree separated one genotype from that
community. The entropy for the reticulated network was lower for all populations
except for the Coast population.



# Discussion {-}

We have presented here new model-free tools for the analysis of genomic data
from clonal populations. 

Creating structures like minimum spanning networks and dendrograms allow
researchers to distill the most important information from large distance
matrices, revealing patterns that could support hypotheses of differentiation or
the lack thereof. Bifurcating dendrograms are most familiar to biologists as the
interpretation of them is straightforward and bootstrap confidence values can
easily be obtained due to the basic structure of the tree. Minimum spanning
networks allow for a different view into populations, where samples themselves
can be treated as internal nodes connecting other samples, which could
effectively describe populations sampled through time. The drawback to these is
that there is no clear method for a bootstrap procedure to obtain confidence
intervals.

Reticulate minimum spanning networks are very important for clonal organisms
where a minimum spanning tree would become a chain, implying that the clones
were derived in a progressive and linear fashion. This presents but one
potential scenario for clonal organisms, but does not account for any other
biologically relevant process. Reticulations in the minimum spanning networks
allow for a representation of uncertainty that goes along with clonal organisms.
The current implementation in *poppr* has been successfully used in analyses
such as reconstruction of the *P. ramorum* outbreak in Curry County, OR
[@kamvar2014sudden; @kamvar2015spatial]. Reticulated networks also allow for the
application of graph community detection algorithms such as the infoMAP
algorithm, [@rosvall2008maps]. As shown in the *P. ramorum* and H3N2 data,
while it is possible to utilize these graph walking algorithms on non-reticulate
minimum spanning trees, the results derived from these are limited to explain
populations derived from serial cloning events. 

- bootstrapping methods encourage future developers to write distance implementations in common format
- moving towards open source, modular tools is the direction that population genetics and plant pathology needs to go. 


# Figure and Table Legends {-}

## Figure 1 {-}

Graphical representation of three different clustering algorithms collapsing
multilocus genotypes for 12 SSR loci from *Phytophthora infestans* representing
18 clonal lineages. The horizontal axis is Bruvo's genetic distance assuming the
genome addition model. The vertical axis represents the number of multilocus
lineages observed. Each point shows the threshold at which one would observe a
given number of multilocus genotypes. The horizontal black line represents 18
multilocus genotypes and vertical dashed lines mark the thresholds used to
collapse the multilocus genotypes int 18 multilocus lineages.

## Figure 2 {-}

(**A-B**) Minimum spanning networks of the hemagglutinin (HA) segment of H3N2
viral DNA from the *adegenet* package representing flu epidemics from 2001 to
2006 with (**B**) and without (**A**) reticulations [@Jombart_2008;
@jombart2010discriminant]. Each node represents a unique multilocus genotype,
colors represent epidemic year, and edge color represents absolute genetic
distance. (**C**) Shannon entropy values for population assignments compared
with communities determined by the infoMAP algorithm on (**A**) and (**B**).
(**D**) Graphic reproduced from @jombart2010discriminant showing that the 2006
epidemic does not cluster neatly with the other years.

## Table 1 {-}

Contingency table comparing multilocus lineages assigned based on average
neighbor clustering (columns) vs. multilocus lineages defined in
@li2013efficient and @lees2006novel.

# References {-}