%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% This is just an example/guide for you to refer to when submitting manuscripts to Frontiers, it is not mandatory to use frontiers.cls nor frontiers.tex  %
% This will only generate the Manuscript, the final article will be typeset by Frontier after acceptance.                                                 %
%                                                                                                                                                         %
% When submitting your files, remember to upload this *tex file, the pdf generated with it, the *bib file (if bibliography is not within the *tex) and all the figures.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% Version 3.0 Generated 2014/12/19 %%%
%%% You will need to have the following packages installed: datetime, fmtcount, etoolbox, fcprefix, which are normally inlcuded in WinEdt. %%%
%%% In http://www.ctan.org/ you can find the packages and how to install them, if necessary. %%%

\documentclass{frontiersSCNS} % for Science, Engineering and Humanities and Social Sciences articles
%\documentclass{frontiersHLTH} % for Health articles
%\documentclass{frontiersFPHY} % for Physics articles

%\setcitestyle{square}
\usepackage{url,lineno}
\linenumbers


% BELOW TAKEN FROM rticles plos template
%
% amsmath package, useful for mathematical formulas
\usepackage{amsmath}
% amssymb package, useful for mathematical symbols
\usepackage{amssymb}

% hyperref package, useful for hyperlinks
\usepackage{hyperref}

% graphicx package, useful for including eps and pdf graphics
% include graphics with the command \includegraphics
\usepackage{graphicx}

% Sweave(-like)
\usepackage{fancyvrb}
\DefineVerbatimEnvironment{Sinput}{Verbatim}{fontshape=sl}
\DefineVerbatimEnvironment{Soutput}{Verbatim}{}
\DefineVerbatimEnvironment{Scode}{Verbatim}{fontshape=sl}
\newenvironment{Schunk}{}{}
\DefineVerbatimEnvironment{Code}{Verbatim}{}
\DefineVerbatimEnvironment{CodeInput}{Verbatim}{fontshape=sl}
\DefineVerbatimEnvironment{CodeOutput}{Verbatim}{}
\newenvironment{CodeChunk}{}{}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

\usepackage{color}

% Below is from frontiers
%
\bibliographystyle{frontiersinSCNS}
% Use doublespacing - comment out for single spacing
%\usepackage{setspace}
%\doublespacing


% Leave a blank line between paragraphs instead of using \\



\def\keyFont{\fontsize{8}{11}\helveticabold }

%% ** EDIT HERE **
%% PLEASE INCLUDE ALL MACROS BELOW

%% END MACROS SECTION



 \def\Authors{
  Zhian N. Kamvar\,\textsuperscript{1},
  Jonah C. Brooks\,\textsuperscript{2},
  Niklaus J. Grunwald\,\textsuperscript{1,3*}}
% \\
\def\Address{
  \textsuperscript{1} Botany and Plant Pathology, Oregon State University,  Corvallis,  OR,  USA\\
  \textsuperscript{2} College of Electrical Engineering and Computer Science, Oregon State University,  Corvallis,  OR,  USA\\
  \textsuperscript{3} Horticultural Crops Research Laboratory, USDA-Agricultural Research Service,  Corvallis,  OR,  USA\\
}

  
  \def\firstAuthorLast{Kamvar {et~al.}}
  
  
  \def\corrAuthor{Niklaus J. Grunwald}\def\corrAddress{Horticultural Crops Research Laboratory USDA ARS\\3420 NW Orchard Ave.\\Corvallis, OR, 97330}\def\corrEmail{\href{mailto:grunwaln@science.oregonstate.edu}{\nolinkurl{grunwaln@science.oregonstate.edu}}}
  


\begin{document}
% \inputencoding{utf8}
\onecolumn
\firstpage{1}

\title[Novel tools for population genetics]{Novel tools for analyzing genome-wide data of clonal populations}
\author[\firstAuthorLast]{\Authors}
\address{}
\correspondance{}
\extraAuth{}% If there are more than 1 corresponding author, comment this line and uncomment the next one.
%\extraAuth{corresponding Author2 \\ Laboratory X2, Institute X2, Department X2, Organization X2, Street X2, City X2 , State XX2 (only USA, Canada and Australia), Zip Code2, X2 Country X2, email2@uni2.edu}
\topic{}% If your article is part of a Research Topic, please indicate here which.
\maketitle

\begin{abstract}

To gain a detailed understanding of plant microbe interactions, adaptation of
pathogens to hosts, develop reactive plant breeding programs, and to deploy R
genes effectively, knowledge of the population dynamics and evolutionary history
of populations is crucial. With the advent of high throughput sequencing
technologies, obtaining genomic sequences for representative populations has
become easier than ever before. A move towards open, reproducible science has
provided impetus for developing population genetic analysis tools in R. We
previously contributed the R package \textit{poppr} specifically addressing
issues with analysis of clonal populations. In this paper we provide several
significant extensions to \textit{poppr} with a focus on large, genome wide SNP
data. Specifically, we provide analyses across any level of hierarchies, a new
function to define clone boundaries we call \texttt{mlg.filter} allowing for
inspection and definition of what is a clonal lineage, and the index of
association for reduced representation genomic data, and modular bootstrapping
of any genetic distance.

\tiny
 \keyFont{ \section{Keywords:} clonality, population genetics, bootstrap, open source}
\end{abstract}

\section*{Introduction}\label{introduction}
\addcontentsline{toc}{section}{Introduction}

To paraphrase Dobzhansky, nothing in the field of plant-microbe
interactions makes sense except in the light of population genetics
(Dobzhansky, 1973). Genetic forces such as selection and drift act on
alleles in a population. Thus, a true understanding of how plant
pathogens evolve and adapt to crops, fungicides, or other factors, can
only emerge in the context of population level phenomena given the
demographic history of populations (McDonald and Linde, 2002; Grunwald
and Goss, 2011; Milgroom et al., 1989). The field of population
genetics, in the era of whole genome resequencing, provides
unprecedented power to describe the evolutionary history and population
processes that drive coevolution between pathogens and hosts. This
powerful field thus critically enables effective deployment of R genes,
design of pathogen informed plant resistance breeding programs, and
implementation of fungicide rotations that minimize emergence of
resistance.

Most computational tools for population genetics are based on concepts
developed for sexual model organisms. Populations that reproduce
clonally or are polyploid are thus difficult to characterize using
classical population genetic tools because theoretical assumptions
underlying the theory are violated. Yet, many plant pathogen populations
are at least partially clonal if not completely clonal (Milgroom, 1996;
Anderson and Kohn, 1995). Thus, development of tools for analysis of
clonal or polyploid populations is needed.

Genotyping by sequencing and whole genome resequencing provide the
unprecedented ability to identify \textgreater{}1,000 single nucleotide
polymorphisms (SNPs) in populations (Elshire et al., 2011; Luikart et
al., 2003; Davey et al., 2011). Availability of these large SNP data
sets provides new challenges for data analysis. For example, it is not
clear what a clone is in large SNP data where the chance of observing
variation at a given SNP locus within independent samples of the same
clone are substantial enough that novel tools for definition of clone
boundaries are required. With traditional marker data (e.g., SSR, AFLP)
a clone was typically defined as a unique multilocus genotype (MLG).
However, with large SNP data a measure of genetic distance is required
to define the boundary of an MLG (e.g., clone) or the boundaries of a
clonal lineage. Definition of a clone is further complicated by the
presence of missing data that is typical for reduced representation
libraries used in GBS or genome re-sequencing. Are two individuals that
are identical for all observed SNPs different if one additional locus
has missing data?

Another major issues with large SNP data is computational efficiency.
Analyses that resample a data set or rely on MCMC are computationally
slow for large data and thus are preferentially implemented in a low
level language such as C rather than in interpreted languages such as R.

The research community using the R statistical and computing language
(RCoreTeam, 2013) has developed a plethora of new resources for
population genetic analysis (Paradis, 2010; Jombart, 2008). Recently, we
introduced the R package \emph{poppr} specifically developed for
analysis of clonal populations (Kamvar et al., 2014b). \emph{Poppr}
previously introduced several novel features including the ability to
conduct a hierarchical analysis across unlimited hierarchies, test for
linkage association, graph minimum spanning networks or provide
bootstrap support for Bruvo's distance in resulting trees. It was well
received by the community, garnering 14 citations in its first year of
publication.

In version 1.1, to address difficulties with handling hierarchical and
mutlilocus genotypic metadata, a new S4 object called ``genclone'' was
defined to expand the genind object of \emph{adegenet}. The genclone
object formalized the definitions of multilocus genotypes and population
hierarchies by adding two slots called mlg and hierarchy that carried a
numeric vector and a data frame, respectively. These new slots allow for
increased efficiency and ease of use by allowing these metadata to
travel with the genetic data. The addition of the population hierarchies
has proved to be advantageous enough that they have recently been
adopted into the more central adegenet package (Jombart, 2008).

In version 1, \emph{poppr} was appropriate for traditional markers
systems, but not well suited to population genomic data resulting from
high throughput sequencing methods. The raw size of these data made it
difficult to conduct traditional analyses. Here, we introduce
\emph{poppr} 2.0, which provides a significant update to \emph{poppr}
including novel tools for analysis of clonal populations specifically
for large SNP data. Significant novel tools include functions for
calculating clone boundaries and collapsing individuals into
user-specified clones based on genetic distance, sliding window
analyses, genotype accumulation curves, reticulations in minimum
spanning networks, and bootstrapping for any genetic distance.

\section*{Materials and Methods}\label{materials-and-methods}
\addcontentsline{toc}{section}{Materials and Methods}

\subsection*{Clonal identification}\label{clonal-identification}
\addcontentsline{toc}{subsection}{Clonal identification}

As highlighted in previous work, clone correction is an important
component of population genetic analysis of organisms that have cryptic
growth or are known to reproduce asexually (Kamvar et al., 2014b;
Milgroom, 1996; Grünwald et al., 2003). This method removes bias that
would otherwise affect metrics that rely on allele frequencies. It was
initially designed for data with only a handful of markers. With the
advent of large-scale sequencing and reduced-representation libraries,
it has become easier to sequence tens of thousands of markers from
hundreds of individuals (Elshire et al., 2011; Davey et al., 2011; Davey
and Blaxter, 2010). With this larger number of markers, the genetic
resolution is much greater, but the chance of genotyping error is also
greatly increased (Mastretta-Yanes et al., 2015). Taking this fact and
occasional somatic mutations into account, it would be impossible to
separate true clones from independent individuals by just comparing what
multilocus genotypes are different. We introduce a new method for
collapsing unique multilocus genotypes determined by naive string
comparison into multilocus lineages utilizing any genetic distance given
three different clustering algorithms: farthest neighbor, nearest
neighbor, and UPGMA (average neighbor) (Sokal, 1958).

The clustering algorithms act on a distance matrix that is either
provided by the user or generated via a function that will calculate a
distance from genclone objects such as \texttt{bruvo.dist}, which in
particular applies to any level of ploidy (Bruvo et al., 2004). All
algorithms have been implemented in C and utilize the OpenMP framework
for optional parallel processing (Dagum and Menon, 1998). Default is the
conservative farthest neighbor algorithm, which will only cluster
samples together if all samples in the cluster are at a distance less
than the given threshold. By contrast, the nearest neighbor algorithm
will have a chaining effect that will cluster samples akin to adding
links on a chain where a sample can be included in a cluster if all of
the samples have at least one connection below a given threshold. The
UPGMA, or average neighbor clustering algorithm is the one most familiar
to biologists as it is often used to generate preliminary ultra-metric
trees based on genetic distance. This algorithm will cluster by creating
a representative sample per cluster and joining clusters if these
representative samples are closer than the given threshold.

\subsection*{\texorpdfstring{Demonstration data: \emph{P.
infestans}}{Demonstration data: P. infestans}}\label{demonstration-data-p.-infestans}
\addcontentsline{toc}{subsection}{Demonstration data: \emph{P.
infestans}}

We utilize data from the microbe \emph{Phytophthora infestans} to show
how the \texttt{mlg.filter} function collapses multilocus genotypes with
Bruvo's distance assuming a genome addition model (Bruvo et al., 2004).
\emph{P. infestans} is the causal agent of potato late blight
originating from Mexico and spread to Europe in the mid 19th century
(Goss et al., 2014; Li et al., 2013; Lees et al., 2006). \emph{P.
infestans} reproduces both clonally and sexually. The clonal lineages of
\emph{P. infestans} have been formally defined into 18 separate clonal
lineages using a combination of various molecular methods including AFLP
and microsatellite markers (Lees et al., 2006). For these data, we used
\texttt{mlg.filter} to detect all of the distance thresholds at which 18
multilocus lineages would be resolved. We used these thresholds to
define multilocus lineages and create contingency tables and dendrograms
to determine how well the multilocus lineages were detected.

\subsection*{Demonstration data: Simulated
data}\label{demonstration-data-simulated-data}
\addcontentsline{toc}{subsection}{Demonstration data: Simulated data}

We utilized simulated data constructed using the \texttt{glSim} function
in adegenet (Jombart and Ahmed, 2011) to obtain a SNP data set for
demonstration. Two diploid data sets were created, each with 10k SNPs
(25\% structured into two groups) and 200 samples with 10 ancestral
populations of even sizes. Clones were created in one data set by
marking each sample with a unique identifier and then randomly sampling
with replacement. It is well documented that reduced- representation
sequencing can introduce several erroneous calls and missing data
(Mastretta-Yanes et al., 2015). To reflect this, we mutated SNPs at a
rate of 10\% and inserted an average of 10\% missing data for each
sample after clones were created, ensuring that no two sequences were
alike. The number of mutations and missing data per sample were
determined by sampling from a poisson distribution with
\(\lambda = 1000\). After pooling, 20\% of the data set was randomly
sampled for analysis. Genetic distance was obtained with the function
\texttt{bitwise.dist}, which calculates the fraction of different sites
between samples, counting missing data as equivalent in comparison.

All three filtering algorithms were run with a threshold of 1, returning
a numeric vector of length \(n - 1\) where each element represented a
threshold at which two samples/clusters would join. Since each data set
would have varying distances between samples, the clonal boundary
threshold was defined as the midpoint of the largest gap between two
thresholds that collapsed less than 50\% of the data.

\subsection*{Index of association}\label{index-of-association}
\addcontentsline{toc}{subsection}{Index of association}

The index of association (\(I_A\)) is a measure of multilocus linkage
disequilibrium that is most often used to detect clonal reproduction
within organisms that have the ability to reproduce via sexual or
asexual processes (Brown et al., 1980; Smith et al., 1993; Milgroom,
1996). It was standardized in 2001 as \(\bar{r}_d\) by Agapow and Burt
(2001) to address the issue of scaling with increasing number of loci.
This metric is typically applied to traditional dominant and co-dominant
markers such as AFLPs, SNPs, or microsatellite markers. With the advent
of high throughput sequencing, SNP data is now available in in a
genome-wide context and in very large matrices including thousands of
SNPs. Thus, the likelihood of finding mutations within two individuals
of a given clone increases and tools are needed for defining clone
boundaries. For this reason, we devised two approaches using the index
of association for large numbers of markers typical for population
genomic studies.

The first approach is a sliding window approach that would utilize the
position of markers in the genome to calculate \(\bar{r}_d\) among any
number of SNPs found within the windowed region. It is important that
this calculation utilize \(\bar{r}_d\) as the number of loci will be
different within each window (Agapow and Burt, 2001). This approach
would be suited for a quick calculation of linkage disequilibrium across
the genome that can detect potential hotspots of LD that could be
investigated further with more computationally intensive methods
assuming that the number of samples \textless{}\textless{} the number of
loci.

As it would necessarily focus on loci within a short section of the
genome that may or may not be recombining, a sliding window approach
would not be good for utilizing \(\bar{r}_d\) as a test for clonal
reproduction. A remedy for this is to randomly sample \(m\) loci,
calculate \(\bar{r}_d\) and repeat \(r\) times, creating a distribution
of expected values of \(\bar{r}_d\).

\subsection*{Population Strata and
Hierarchies}\label{population-strata-and-hierarchies}
\addcontentsline{toc}{subsection}{Population Strata and Hierarchies}

Assessments of population structure through methods such as hierarchical
\(F_{st}\) and AMOVA benefit greatly from multiple levels of population
definition (Linde et al., 2002; Everhart and Scherm, 2015; Gr{ü}nwald
and Hoheisel, 2006). With clonal organisms, basic practice has been to
clone-censor data to avoid downward bias in diversity due to duplicated
genotypes that may or may not represent different samples (Milgroom,
1996). Data structures for population genetic data mostly allow for only
one level of hierarchical definition. The impetus was placed on the
researchers to provide the population hierarchies for every step of the
analysis. In poppr version 1.1, the hierarchy slot was introduced to
allow unlimited population hierarchies or stratifications to travel with
the data. In practice, it is stored as a data frame where each column
represents a separate hierarchical level. This is then used to set the
population factor of the data by supplying a hierarchical formula
containing one or more column names of the data frame in the hierarchy
slot. This functionality, developed in \emph{poppr}, has been moved to
the \emph{adegenet} package in version 2.0 and the slot and methods have
been renamed to \texttt{strata}.

\subsection*{Genotype Accumulation
Curve}\label{genotype-accumulation-curve}
\addcontentsline{toc}{subsection}{Genotype Accumulation Curve}

Analysis of population genetics of clonal organisms often borrows from
ecological methods such as analysis of diversity within populations
(Milgroom, 1996; Arnaud-Hanod et al., 2007; Grünwald et al., 2003). When
choosing markers for analysis, it is important to make sure that the
observed diversity in your sample will not appreciably increase if an
additional marker is added (Arnaud-Hanod et al., 2007). This concept is
analogous to a species accumulation curve, obtained by rarefaction. The
genotype accumulation curve in \emph{poppr} is implemented by randomly
sampling \(x\) loci and counting the number of observed MLGs. This
repeated \(r\) times for 1 locus up to \(n-1\) loci, creating \(n-1\)
distributions of observed MLGs.

\subsection*{Minimum Spanning Networks with
Reticulation}\label{minimum-spanning-networks-with-reticulation}
\addcontentsline{toc}{subsection}{Minimum Spanning Networks with
Reticulation}

In its original iteration, \emph{poppr} introduced minimum spanning
networks that were based on the \emph{igraph} function
\texttt{minimum.spanning.tree} (Csardi and Nepusz, 2006). This algorithm
produces a minimum spanning tree with no reticulations, which is common
and expected for clonal organisms. In other minimum spanning network
programs, reticulation is obtained by calculating the minimum spanning
tree several times and returning the set of all edges included in the
trees. Due to the way \emph{igraph} has implemented Prim's algorithm, it
is not possible to utilize this strategy, thus we implemented a C
function to walk the space of minimum spanning trees based on genetic
distance to connect groups of nodes with edges of equal weight.

To demonstrate the utility of minimum spanning networks with
reticulation, we used two clonal data sets: H3N2 flu virus data from the
\emph{adegenet} package using years of each epidemic as the population
factor, and \emph{Phytophthora ramorum} data from Nurseries and Oregon
forests (Jombart et al., 2010; Kamvar et al., 2014a). Minimum spanning
networks were created with and without reticulation using the
\emph{poppr} functions \texttt{diss.dist} and \texttt{bruvo.msn} for the
H3N2 and \emph{P. ramorum} data, respectively (Kamvar et al., 2014b;
Bruvo et al., 2004). To detect mlg clusters, the infoMAP community
detection algorithm was applied with 10,000 trials as implemented in the
R package \emph{igraph} version 0.7.1 utilizing genetic distance as edge
weights and number of samples in each MLG as vertex weights (Csardi and
Nepusz, 2006; Rosvall and Bergstrom, 2008).

To evaluate the results, we compared the number and size of resulting
communities as we expect a highly clonal organism with low genetic
diversity to result in a few, large communities. We also created
contingency tables of the community assignments with the defined
populations and used those to calculate entropy using Shannon's index
with the function \texttt{diversity} from the R package \emph{vegan}
version 2.2-1 (Oksanen et al., 2015; Shannon, 2001). A low entropy
indicates presence of a few large communities whereas high entropy
indicates presence of many small communities.

\subsection*{Bootstrapping}\label{bootstrapping}
\addcontentsline{toc}{subsection}{Bootstrapping}

Calculating genetic distance for among samples and populations is very
important method for assessing population differentiation through
methods such as \(F_{st}\), AMOVA, and Mantel tests. Confidence in
distance metrics is related to the confidence in the markers to
accurately represent the diversity of the data. Especially true with
microsatellite markers, a single hyper-diverse locus can make a
population appear to have more diversity based on genetic distance.
Using a bootstrapping procedure of randomly sampling loci with
replacement when calculating a distance matrix gives confidence in
hierarchical clustering. Because genetic data in a genind object is
represented as a matrix with samples in rows and alleles in columns,
bootstrapping is a non-trivial task as all alleles in a single locus
need to be sampled together. To remedy this, we have created an internal
S4 class called ``bootgen'', which extends the internal ``gen'' class
from adegenet. This class can be created from any genind, genclone, or
genpop object, and allows loci to be sampled with replacement. To
further facilitate bootstrapping, a function called aboot, which stands
for ``any boot'', is introduced that will bootstrap any genclone,
genind, or genpop object with any genetic distance that can be
calculated from it.

\section*{Results}\label{results}
\addcontentsline{toc}{section}{Results}

\subsection*{Clonal identification}\label{clonal-identification-1}
\addcontentsline{toc}{subsection}{Clonal identification}

For the \emph{P. infestans} population, the three algorithms were able
to detect 18 multilocus lineages at different distance thresholds (Fig.
1). Contingency tables between the described multilocus genotypes and
the genotypes defined by distance show that most of the 18 lineages were
resolved, except for US-8, which is polytomic (Table 1). Out of the 100
simulations run, we found that across all methods, detection of
duplicated samples had \(\sim\) 98\% true positive fraction and \(\sim\)
0.8\% false positive fraction indicating that this method is robust to
simulated populations.

\subsection*{Minimum Spanning Network with
Reticulation}\label{minimum-spanning-network-with-reticulation}
\addcontentsline{toc}{subsection}{Minimum Spanning Network with
Reticulation}

The infoMAP algorithm revealed 63 communities with a maximum community
size of 77 and \(H = 3.56\) for the reticulate network of the H3N2 data
and 117 communities with a maximum community size of 26 and \(H = 4.65\)
for the minimum spanning tree. The entropy across years was greatly
decreased for all populations with the reticulate network compared to
the minimum spanning tree (Fig 2).

Graph walking of the reticulated minimum spanning network of \emph{P.
ramorum} by the infoMAP algorithm revealed 16 communities with a maximum
community size of 13 and \(H = 2.60\). The un-reticulated minimum
spanning tree revealed 20 communities with a maximum community size of 7
and \(H = 2.96\). In the ability to predict Hunter Creek as belonging to
a single community, the reticulated network was successful whereas the
minimum spanning tree separated one genotype from that community. The
entropy for the reticulated network was lower for all populations except
for the Coast population (supplementary information).

\subsection*{Example: Bootstrap population
dendrogram}\label{example-bootstrap-population-dendrogram}
\addcontentsline{toc}{subsection}{Example: Bootstrap population
dendrogram}

The following code example was used to demonstrate calculating a
dendrogram with bootstrap support using the function \texttt{aboot} on
the data set \texttt{microbov} from the \emph{adegenet} package
(Jombart, 2008; Lalo{ë} et al., 2007). The resulting dendrogram shows
bootstrap support values \(>50\%\) (Fig. 3).

\begin{CodeChunk}
\begin{CodeInput}
library("poppr")
data("microbov", package = "adegenet") 
strata(microbov) <- data.frame(other(microbov)) 
setPop(microbov) <- ~coun/spe/breed 
bov_pop <- genind2genpop(microbov, quiet = TRUE) 

set.seed(20150428)
pop_tree <- aboot(bov_pop, sample = 1000, cutoff = 50, quiet = TRUE)
\end{CodeInput}
\end{CodeChunk}

\subsection*{Example: Genotype accumulation
curve}\label{example-genotype-accumulation-curve}
\addcontentsline{toc}{subsection}{Example: Genotype accumulation curve}

The following code example demonstrates the genotype accumulation curve
for data from Everhart and Scherm (2015) showing that these data reach a
small plateau, indicating that there are enough markers such that adding
more markers to the analysis will not create very many new genotypes
(Fig. 4).

\begin{CodeChunk}
\begin{CodeInput}
library("poppr")
library("ggplot2")
data("monpop", package = "poppr")

set.seed(20150428)
genotype_curve(monpop, sample = 1000, quiet = TRUE)
p <- last_plot() + theme_bw()   # get the last plot
p + geom_smooth(aes(group = 1)) # plot with a trendline
\end{CodeInput}
\end{CodeChunk}

\section*{Discussion}\label{discussion}
\addcontentsline{toc}{section}{Discussion}

We have presented here new model-free tools for the analysis of clonal
populations with emphasis on genomic-scale data. Especially important is
\texttt{mlg.filter}, which

Creating structures like minimum spanning networks and dendrograms allow
researchers to distill the most important information from large
distance matrices, revealing patterns that could support hypotheses of
differentiation or the lack thereof. Bifurcating dendrograms are most
familiar to biologists as the interpretation of them is straightforward
and bootstrap confidence values can easily be obtained due to the basic
structure of the tree. Minimum spanning networks allow for a different
view into populations, where samples themselves can be treated as
internal nodes connecting other samples, which could effectively
describe populations sampled through time. The drawback to these is that
there is no clear method for a bootstrap procedure to obtain confidence
intervals.

Reticulate minimum spanning networks are very important for clonal
organisms where a minimum spanning tree would become a chain, implying
that the clones were derived in a progressive and linear fashion. This
presents but one potential scenario for clonal organisms, but does not
account for any other biologically relevant process. Reticulations in
the minimum spanning networks allow for a representation of uncertainty
that goes along with clonal organisms. The current implementation in
\emph{poppr} has been successfully used in analyses such as
reconstruction of the \emph{P. ramorum} epidemic in Curry County, OR
(Kamvar et al., 2014a, 2015). Reticulated networks also allow for the
application of graph community detection algorithms such as the infoMAP
algorithm (Rosvall and Bergstrom, 2008). As shown in the \emph{P.
ramorum} and H3N2 data, while it is possible to utilize these graph
walking algorithms on non-reticulate minimum spanning trees, the results
derived from these are limited to explain populations derived from
serial cloning events.

\begin{itemize}
\itemsep1pt\parskip0pt\parsep0pt
\item
  bootstrapping methods encourage future developers to write distance
  implementations in common format
\item
  moving towards open source, modular tools is the direction that
  population genetics and plant pathology needs to go.
\end{itemize}

\section*{Figures and Tables}\label{figures-and-tables}
\addcontentsline{toc}{section}{Figures and Tables}

\subsection*{Figure 1}\label{figure-1}
\addcontentsline{toc}{subsection}{Figure 1}

\begin{CodeChunk}

\includegraphics[width=80mm,height=80mm]{poppr_frontiers_files/figure-latex/pinf_data-1} \end{CodeChunk}

\subsection*{Figure 2}\label{figure-2}
\addcontentsline{toc}{subsection}{Figure 2}

\includegraphics{poppr_frontiers_files/custom_figures/Figure-2.png}

\subsection*{Figure 3}\label{figure-3}
\addcontentsline{toc}{subsection}{Figure 3}

\begin{CodeChunk}

\includegraphics[width=80mm,height=80mm]{poppr_frontiers_files/figure-latex/microboot2-1} \end{CodeChunk}

\subsection*{Figure 4}\label{figure-4}
\addcontentsline{toc}{subsection}{Figure 4}

\begin{CodeChunk}

\includegraphics[width=80mm,height=80mm]{poppr_frontiers_files/figure-latex/monilinia-1} \end{CodeChunk}

\subsection*{Table 1}\label{table-1}
\addcontentsline{toc}{subsection}{Table 1}

\begin{table}[ht]
\centering
\begin{tabular}{ccccccccccccccccccc}
  \hline
 & 3 & 4 & 5 & 6 & 8 & 10 & 12 & 15 & 16 & 17 & 18 & 20 & 21 & 22 & 24 & 25 & 27 & 28 \\ 
  \hline
B & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & 1 & . & . \\ 
  C & . & . & . & . & . & . & . & . & . & . & . & . & . & . & 1 & . & . & . \\ 
  D.1 & . & . & . & . & . & . & . & . & . & . & . & . & . & 1 & . & . & . & . \\ 
  D.2 & . & . & . & . & . & . & . & . & . & . & . & . & . & 1 & . & . & . & . \\ 
  EU-13 & . & . & . & . & . & . & . & . & 1 & . & . & . & . & . & . & . & . & . \\ 
  EU-4 & . & . & . & . & . & . & . & . & . & 1 & . & . & . & . & . & . & . & . \\ 
  EU-5 & . & . & . & . & . & . & . & . & . & . & 2 & . & . & . & . & . & . & . \\ 
  EU-8 & . & . & . & . & . & . & 1 & . & . & . & . & . & . & . & . & . & . & . \\ 
  US-11 & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & 2 \\ 
  US-12 & . & 1 & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . \\ 
  US-14 & . & . & . & . & . & 1 & . & . & . & . & . & . & . & . & . & . & . & . \\ 
  US-17 & . & . & . & . & . & . & . & . & . & . & . & 1 & . & . & . & . & . & . \\ 
  US-20 & 2 & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . \\ 
  US-21 & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & . & 2 & . \\ 
  US-22 & . & . & . & . & . & . & . & . & . & . & . & . & 2 & . & . & . & . & . \\ 
  US-23 & . & . & . & . & . & . & . & 3 & . & . & . & . & . & . & . & . & . & . \\ 
  US-24 & . & . & . & . & 3 & . & . & . & . & . & . & . & . & . & . & . & . & . \\ 
  US-8 & . & . & 1 & 1 & . & 2 & . & . & . & . & . & . & . & . & . & . & . & . \\ 
   \hline
\end{tabular}
\end{table}

\section*{Figure and Table Legends}\label{figure-and-table-legends}
\addcontentsline{toc}{section}{Figure and Table Legends}

\subsection*{Figure 1}\label{figure-1-1}
\addcontentsline{toc}{subsection}{Figure 1}

Graphical representation of three different clustering algorithms
collapsing multilocus genotypes for 12 SSR loci from \emph{Phytophthora
infestans} representing 18 clonal lineages. The horizontal axis is
Bruvo's genetic distance assuming the genome addition model. The
vertical axis represents the number of multilocus lineages observed.
Each point shows the threshold at which one would observe a given number
of multilocus genotypes. The horizontal black line represents 18
multilocus genotypes and vertical dashed lines mark the thresholds used
to collapse the multilocus genotypes int 18 multilocus lineages.

\subsection*{Figure 2}\label{figure-2-1}
\addcontentsline{toc}{subsection}{Figure 2}

(\textbf{A-B}) Minimum spanning networks of the hemagglutinin (HA)
segment of H3N2 viral DNA from the \emph{adegenet} package representing
flu epidemics from 2001 to 2006 with (\textbf{B}) and without
(\textbf{A}) reticulations (Jombart, 2008; Jombart et al., 2010). Each
node represents a unique multilocus genotype, colors represent epidemic
year, and edge color represents absolute genetic distance. (\textbf{C})
Shannon entropy values for population assignments compared with
communities determined by the infoMAP algorithm on (\textbf{A}) and
(\textbf{B}). (\textbf{D}) Graphic reproduced from Jombart et al. (2010)
showing that the 2006 epidemic does not cluster neatly with the other
years.

\subsection*{Figure 3}\label{figure-3-1}
\addcontentsline{toc}{subsection}{Figure 3}

UPGMA dendrogram generated from Nei's gentic distance on 15 breeds of
\emph{Bos taurus} (BT) or \emph{Bos indicus} (BI) from Africa (AF) or
France (FR). These data are from Lalo{ë} et al. (2007). Node labels
represent bootstrap support \(>50\%\) out of 1,000 bootstrap replicates.

\subsection*{Figure 4}\label{figure-4-1}
\addcontentsline{toc}{subsection}{Figure 4}

Genotype accumulation curve for 694 isolates of the peach brown rot
pathogen, \emph{Monilinia fructicola} genotyped over 13 loci from
Everhart and Scherm (2015). The horizontal axis represents the number of
loci randomly sampled without replacement up to \(n - 1\) loci, the
vertical axis shows the number of multilocus genotypes observed, up to
262, the number of unique multilocus genotypes in the data set. The red
dashed line represents 90\% of the total observed multilocus genotypes.
A trendline (blue) has been added using the \emph{ggplot2} function
\texttt{stat\_smooth}.

\subsection*{Table 1}\label{table-1-1}
\addcontentsline{toc}{subsection}{Table 1}

Contingency table comparing multilocus lineages assigned based on
average neighbor clustering (columns) vs.~multilocus lineages defined in
Li et al. (2013) and Lees et al. (2006).

\section*{References}\label{references}
\addcontentsline{toc}{section}{References}

Agapow, P.-M., and Burt, A. (2001). Indices of multilocus linkage
disequilibrium. \emph{Molecular Ecology Notes} 1, 101--102.
doi:\href{http://dx.doi.org/10.1046/j.1471-8278.2000.00014.x}{10.1046/j.1471-8278.2000.00014.x}.

Anderson, J. B., and Kohn, L. M. (1995). Clonality in soilborne,
plant-pathogenic fungi. \emph{Annual review of phytopathology} 33,
369--391.

Arnaud-Hanod, S., Duarte, C. M., Alberto, F., and Serr{ã}o, E. A.
(2007). Standardizing methods to address clonality in population
studies. \emph{Molecular Ecology} 16, 5115--5139.

Brown, A., Feldman, M., and Nevo, E. (1980). MULTILOCUS sTRUCTURE oF
nATURAL pOPULATIONS oF \emph{Hordeum spontaneum}. \emph{Genetics} 96,
523--536. Available at:
\url{http://www.genetics.org/content/96/2/523.abstract}.

Bruvo, R., Michiels, N. K., D'Souza, T. G., and Schulenburg, H. (2004).
A simple method for the calculation of microsatellite genotype distances
irrespective of ploidy level. \emph{Molecular Ecology} 13, 2101--2106.

Csardi, G., and Nepusz, T. (2006). The igraph software package for
complex network research. \emph{InterJournal} Complex Systems, 1695.
Available at: \url{http://igraph.org}.

Dagum, L., and Menon, R. (1998). OpenMP: An industry standard aPI for
shared-memory programming. \emph{Computational Science \& Engineering,
IEEE} 5, 46--55.

Davey, J. W., and Blaxter, M. L. (2010). RADSeq: Next-generation
population genetics. \emph{Briefings in Functional Genomics} 9,
416--423.
doi:\href{http://dx.doi.org/10.1093/bfgp/elq031}{10.1093/bfgp/elq031}.

Davey, J. W., Hohenlohe, P. A., Etter, P. D., Boone, J. Q., Catchen, J.
M., and Blaxter, M. L. (2011). Genome-wide genetic marker discovery and
genotyping using next-generation sequencing. \emph{Nature Reviews
Genetics} 12, 499--510.

Dobzhansky, T. (1973). Nothing in biology makes sense except in the
light of evolution. \emph{The American Biology Teacher} 75, 87--91.

Elshire, R. J., Glaubitz, J. C., Sun, Q., Poland, J. A., Kawamoto, K.,
Buckler, E. S., and Mitchell, S. E. (2011). A robust, simple
genotyping-by-sequencing (gBS) approach for high diversity species.
\emph{PloS one} 6, e19379.

Everhart, S., and Scherm, H. (2015). Fine-scale genetic structure of
\emph{Monilinia fructicola} during brown rot epidemics within individual
peach tree canopies. \emph{Phytopathology} 105, 542--549.

Goss, E. M., Tabima, J. F., Cooke, D. E., Restrepo, S., Fry, W. E.,
Forbes, G. A., Fieland, V. J., Cardenas, M., and Gr{ü}nwald, N. J.
(2014). The irish potato famine pathogen \emph{Phytophthora infestans}
originated in central mexico rather than the andes. \emph{Proceedings of
the National Academy of Sciences} 111, 8791--8796.

Grunwald, N. J., and Goss, E. M. (2011). Evolution and population
genetics of exotic and re-emerging pathogens: Novel tools and
approaches. \emph{Annual Review of Phytopathology} 49, 249--267.

Gr{ü}nwald, N. J., and Hoheisel, G.-A. (2006). Hierarchical analysis of
diversity, selfing, and genetic differentiation in populations of the
oomycete aphanomyces euteiches. \emph{Phytopathology} 96, 1134--1141.

Grünwald, N. J., Goodwin, S. B., Milgroom, M. G., and Fry, W. E. (2003).
Analysis of genotypic diversity data for populations of microorganisms.
\emph{Phytopathology} 93, 738--46. Available at:
\url{http://apsjournals.apsnet.org/doi/abs/10.1094/PHYTO.2003.93.6.738}.

Jombart, T. (2008). Adegenet: a R package for the multivariate analysis
of genetic markers. \emph{Bioinformatics} 24, 1403--1405.
doi:\href{http://dx.doi.org/10.1093/bioinformatics/btn129}{10.1093/bioinformatics/btn129}.

Jombart, T., and Ahmed, I. (2011). Adegenet 1.3-1: New tools for the
analysis of genome-wide sNP data. \emph{Bioinformatics} 27, 3070--3071.

Jombart, T., Devillard, S., and Balloux, F. (2010). Discriminant
analysis of principal components: A new method for the analysis of
genetically structured populations. \emph{BMC genetics} 11, 94.

Kamvar, Z. N., Larsen, M. M., Kanaskie, A. M., Hansen, E. M., and
Grünwald, N. J. (2014a). Sudden\_Oak\_Death\_in\_Oregon\_Forests:
Spatial and temporal population dynamics of the sudden oak death
epidemic in Oregon Forests.
doi:\href{http://dx.doi.org/10.5281/zenodo.13007}{10.5281/zenodo.13007}.

Kamvar, Z. N., Larsen, M. M., Kanaskie, A., Hansen, E., and Grünwald, N.
J. (2015). Spatial and temporal analysis of populations of the sudden
oak death pathogen in oregon forests. \emph{Phytopathology}, in press.

Kamvar, Z. N., Tabima, J. F., and Gr{ü}nwald, N. J. (2014b). Poppr: An r
package for genetic analysis of populations with clonal, partially
clonal, and/or sexual reproduction. \emph{PeerJ} 2, e281.

Lalo{ë}, D., Jombart, T., Dufour, A.-B., and Moazami-Goudarzi, K.
(2007). Consensus genetic structuring and typological value of markers
using multiple co-inertia analysis. \emph{Genetics Selection Evolution}
39, 1--23.

Lees, A., Wattier, R., Shaw, D., Sullivan, L., Williams, N., and Cooke,
D. (2006). Novel microsatellite markers for the analysis of phytophthora
infestans populations. \emph{Plant Pathology} 55, 311--319.

Li, Y., Cooke, D. E., Jacobsen, E., and Lee, T. van der (2013).
Efficient multiplex simple sequence repeat genotyping of the oomycete
plant pathogen phytophthora infestans. \emph{Journal of microbiological
methods} 92, 316--322.

Linde, C., Zhan, J., and McDonald, B. (2002). Population structure of
mycosphaerella graminicola: From lesions to continents.
\emph{Phytopathology} 92, 946--955.

Luikart, G., England, P. R., Tallmon, D., Jordan, S., and Taberlet, P.
(2003). The power and promise of population genomics: From genotyping to
genome typing. \emph{Nature Reviews Genetics} 4, 981--994.

Mastretta-Yanes, A., Arrigo, N., Alvarez, N., Jorgensen, T. H.,
Pi{ñ}ero, D., and Emerson, B. (2015). Restriction site-associated dNA
sequencing, genotyping error estimation and de novo assembly
optimization for population genetic inference. \emph{Molecular ecology
resources} 15, 28--41.

McDonald, B. A., and Linde, C. (2002). The population genetics of plant
pathogens and breeding strategies for durable resistance.
\emph{Euphytica} 124, 163--180.
doi:\href{http://dx.doi.org/10.1023/A:1015678432355}{10.1023/A:1015678432355}.

Milgroom, M. G. (1996). Recombination and the multilocus structure of
fungal populations. \emph{Annual review of phytopathology} 34, 457--477.

Milgroom, M. G., Levin, S. A., and Fry, W. E. (1989). Population
genetics theory and fungicide resistance. \emph{Plant disease
epidemiology} 2, 340--367.

Oksanen, J., Blanchet, F. G., Kindt, R., Legendre, P., Minchin, P. R.,
O'Hara, R. B., Simpson, G. L., Solymos, P., Stevens, M. H. H., and
Wagner, H. (2015). \emph{Vegan: Community ecology package}. Available
at: \url{http://CRAN.R-project.org/package=vegan}.

Paradis, E. (2010). Pegas: an R package for population genetics with an
integrated--modular approach. \emph{Bioinformatics} 26, 419--420.

RCoreTeam (2013). R: A Language and Environment for Statistical
Computing. Available at: \url{http://www.R-project.org/}.

Rosvall, M., and Bergstrom, C. T. (2008). Maps of random walks on
complex networks reveal community structure. \emph{Proceedings of the
National Academy of Sciences} 105, 1118--1123.

Shannon, C. (2001). A mathematical theory of communication. \emph{ACM
SIGMOBILE Mobile Computing and Communications Review} 5, 3--55.
Available at:
\url{http://cm.bell-labs.com/cm/ms/what/shannonday/shannon1948.pdf}.

Smith, J. M., Smith, N. H., O'Rourke, M., and Spratt, B. G. (1993). How
clonal are bacteria? \emph{Proceedings of the National Academy of
Sciences} 90, 4384--4388.
doi:\href{http://dx.doi.org/10.1073/pnas.90.10.4384}{10.1073/pnas.90.10.4384}.

Sokal, R. R. (1958). A statistical method for evaluating systematic
relationships. \emph{Univ Kans Sci Bull} 38, 1409--1438.


\end{document}

